---
title: "Transcriptome of C. difficile Across Susceptible Treatments"
author: "Matthew Jenior"
output: html_document
---

###Abstract


###Introduction
Clostridium difficile is a Gram-positive, sporulating anaerobe which causes a toxin-mediated disease in the gastrointestinal tract of many animals.  Infection by *C. difficile* can cause a spectrum of conditions ranging from asymptomatic colonization, abdominal pain, diarrhea, to the more dangerous symptoms pseudomembraneous colitis and toxin megacolon.  Numerous studies have demonstrated that the gut microbiota limits the capacity of pathogens like *C. difficile* through an attribute known as colonization ().  This collection of indigenous microorganisms that inhabit the gastrointestinal tract positively contributes to numerous aspects of host health aside colonization resistance.  Disruptions to the integrity of the healthy community, through a variety of means including chemotherapy or antibiotic administration, can lead to a complete loss of colonization resistance (Young 2008).  This can become problematic in health-care settings where many patients are not only likely to have factors leading to perturbed gut communities, but may also be exposed to *C. difficile* spores ().  As such, *C. difficile* infections (CDI) have rapidly increased in prevalence over the last decade to become the most prevalent form of hospital-acquired infection infecting >500,000 individuals per year domestically, with ~29,000  of these instances resulting in death (Lessa).

The pathology associated with CDI is mediated by expression of it's toxin genes whose production has many known connections to the bacteria's access to nutirients ().  In vitro, toxin production begins as the culture enters stationary phase ().  It has been shown that high concentrations of easily metabolised carbon sources, such as glucose or amino acids, inhibit toxin production ().  It has also been demonstrated that nutrient limitation activates the master transcriptional regulator CodY ().  Downstream effects of this include ...
Conversely, very little is known about how the perturbed gut microbiota niche landscape is changed in a way to promote *C. difficile* colonization.  A recent study by Theriot at al. demonstrated that the cecal metabolome of antibiotic-treated mice are enriched for established C. difficile growth substrates.  It is therefore not unlikely that other classes of antibiotic that lead to CDI suscpetibility may cause similar changes to the gut nutrient environment.	These findings also lead to the hypothesis that the healthy gut microbiota excludes *C. difficile* based on a preferred nutrient niche.  

A large genome, a wide variety of growth substrates, and a large host range are all strong indications that *C. difficile* is an ecological generalist.  The ability of C. difficile to colonize a tremendous range of community structures also supports the hypothesis that the pathogen could have a relatively large potential niche space it could occupy during infection.



Different classes of antibiotics result in unique community structures, although primarily maintain susceptibility to *C. difficile* colonization.

Presumeably, each of these conditions poses seperate challenges to *C. difficile* 



In this study we observed that not only the disease and lifecycle phenotypes of C. difficile different across antibiotic treatments, but expression of key metabolic pathways was also unique. 
This principle was then extended through using genome-scale metabolic modeling to infer differences in exploited  nutrient niches between the susceptible gut communities.  These findings were supported with in vitro growth assays to demonstrate that C. difficile str. 630 would indeed utilize them for growth to high levels.  Additionally, relative toxin production levels were recapitulated in vitro compared to those in antibiotic treated animals, suggesting alternative nutrient niches may explain differences in disease pathology or severity.  Taken together, it appears that C. difficile has a large potential niche space in the murine gastrointestinal tract and may provided indications of a hierarchy of growth substrate preference in the context of specific competitors.

###Results

Timing of sporulation has been linked to environmental cues for metabolism such as glucose or glycine concentrations ().  With this in mind, differential plating was performed to determine the load of both spores and vegetative cells respectively in each treatment condition.  This was accomplished by first plating on agar media without the primary C. difficile germinant taurocholate to enumerate vegetative cells exclusively.  Immediately following was a 30 minute 65 C heat treatment to eliminate vegetative cells, and plating on media containing taurocholate. 

No difference in vegetative cell load was found in any susceptible condition tested.  However, significantly more (p=0.00012) spores were detected in gnotobiotic mice than in any other the antibiotic treated mice.  Control, untreated mice remained resistant to colonization by C. difficile.








Previous work (mark’s paper) demonstrated that at least one other strain of C. difficile exhibited very little capacity for sporulation 18 hours post-infection while being at maximum bacterial load.  We hypothesized that this timepoint would be optimal for transcriptomic analysis 


+---------------+------------------+-----------------------------------+---------------------------------------+---------------------------+----------------------+
| Antibiotic    | Class            | Bacterial Target                  | Primary Mode of Action                | Route of Administration   | Dosage               |
+===============+==================+===================================+=======================================+===========================+======================+
| Cefoperazone  | Cephalosporin    | Primarily Gram-positive bacteria, | Irreversibly crosslinks bacterial     | Drinking water Ad libitum | 0.5 mg/ml            |
|               | (3rd generation) | with increased activity against   | transpeptidases to peptidoglycan      |                           |                      |
|               |                  | Gram-negative bacteria            | and prevents cell wall synthesis      |                           |                      |
+---------------+------------------+-----------------------------------+---------------------------------------+---------------------------+----------------------+
| Clindamycin   | Lincosamide      | Primarily Gram-positive bacteria, | Protein synthesis inhibition through  | Intraperitoneal injection | 10 mg/kg body weight |
|               |                  | most anaerobic bacteria, and      | binding to the 23s portion of the 50S |                           |                      |
|               |                  | some mycoplasma                   | ribosomal subunit                     |                           |                      |
+---------------+------------------+-----------------------------------+---------------------------------------+---------------------------+----------------------+
| Streptomycin  | Aminoglycoside   | Active against most Gram-negative | Protein synthesis inhibitor through   | Drinking water Ad libitum | 5.0 mg/ml            |
|               |                  | aerobic and facultative anaerobic | binding the 30S portion of the 70S    |                           |                      |
|               |                  | bacilli                           | ribosomal subunit                     |                           |                      |
+---------------+------------------+-----------------------------------+---------------------------------------+---------------------------+----------------------+

Table 1 – Descriptions of antibiotics used during this study


<img src="/Users/mattjenior/Desktop/Repositories/Jenior_Transcriptomics_2015/results/figures/figure_1.pdf" width=800 />

Figure 1 - Sporulation and toxin production by *C. difficile* varies between experimental treatment goups. (A) Experimental timelines for the antibiotic treatments used.  The upper course was used for both Cefoperazone and Streptomycin pretreatments in the drinking water.  The lower couse was used in Clindamycin introperotineal infection prior to *C. difficile* infection. (B) Toxin titer. (C) Vegetative *C. difficile*. (D) *C. diffiicile* spores.

**The gut microbiome in mice influences *C. difficile* sporulation and toxin production**

Sporulation is a critical step for C. difficile to disseminate between hosts. This phenotype has been linked to the local concentration of various nutrient sources and envirnmental signals (citations). We hypothesized that across antibiotic treatment, transcripts for specific nutrient utilization loci would be over represented and signify a focus in nutrient catabolism for the bacterium in the given treatment.

The amount of C. difficile spores quantified in each condition was changedConcentration of spores in each condition was derived through differential plating with and without the primary germinant taurocholate.

Figure 2 - Circos plot of C. difficile genome converage in each of the conditions






<img src="/Users/mattjenior/Desktop/Repositories/Jenior_Transcriptomics_2015/results/figures/figure_4.pdf" width=800 />

Figure 4 - Expression of metabolism genes across the different environments


Genes included in each group are in Supplementary Table 1.


**Genes for alternative forms of carbon metabolism in *C. difficile* are differentially expressed across antibiotic treatment groups**

In each antibiotic treated condition, *C. difficile* 630 demonstrates differential expression of genes in pathways for seperate aspects of carbon metabolism.




These data support the hypothesis that *C. difficile* may inhabit alternative realized niches accross the susceptible microbiomes in colonizes.





<img src="/Users/mattjenior/Desktop/Repositories/Jenior_Transcriptomics_2015/results/figures/figure_5.pdf" width=800 />


**Figure 5 - Bipartite metabolic networks are predictive of *C. difficile* nutrient preference across coolonized environments**

**Metabolic network algorithm predicts important components of *C. difficile*'s nutrient niche in different environments**


**Figure 6 - Nutirents predictive through metabolic modeling**



###Discussion

It seems that the metabolic response of C. difficile is dependent on the context of the community in which it finds itself.  This could be an explanation as to why FMT is such an effective treatment.  If you simultaneously introduce every possible functionality found in a healthy intestinal community, C. difficile cannot compete.  This hypothesis also leads to the idea that efforts to restore coloinization resistance in a targeted way may prove diffciult as C. difficile may be able to aqequately adapt to any new condition short of the diversity of function found in an FMT.




###Materials and Methods
Animal care, antibiotic administration, C. difficile infection, and necropsy  Six-to-eight week-old specific pathogen free (SPF) C57BL/6 mice were obtained from a single breeding colony maintained at the University of Michigan for all experiments. All animal protocols were approved by the University Committee on Use and Care of Animals at the University of Michigan and carried out in accordance with the approved guidelines. Six-to-eight week-old germfree C57BL/6 mice were obtained from a single breeding colony maintained at the University of Michigan for all experiments. All animal protocols were approved by the University Committee on Use and Care of Animals at the University of Michigan and carried out in accordance with the approved guidelines.
	Specified SPF animals were administered one of three antibiotics;  including cefoperazone, streptomycin, or clindamycin (Table 1). Cefoperazone (0.5 mg/ml) and streptomycin (5.0 mg/ml) were both administered in the drinking water for 5 days while clindamycin (10 mg/kg) was given via intraperitoneal injection.  Antibiotic-treated mice were given a brief recovery period (48 hrs for cefoperazone & streptomycin, 24 hrs for clindamycin) with untreated drinking water prior to C. difficile challenge.
	C. difficile strain 630 spores were prepared from a single large batch whose concentration was determined in the week prior to challenge for all experiments (27).  On the day of challenge, 103 C. difficile spores were administered to mice via oral gavage in phosphate-buffered saline (PBS). Subsequent quantitative plating for CFU was performed to ensure correct dosage.
	18 hours following infection, mice was euthanized by carbon dioxide asphyxiation.  Necropsy was then performed and cecal content was split into three small aliquots (~100 μl).  Two were flash frozen immediately for later DNA extraction and toxin titer analysis.  The third aliquot was quickly moved to an anaerobic chamber for CFU quantification.  The remaining content in the ceca (~1 ml) was emptied into a stainless steel mortar in a dry ice/ethanol bath using 1 ml of sterile PBS.  This process was repeated for each mouse within a treatment group to pool content into a single large sample (9 mice).  The content was then finely ground and stored at -80 C for subsequent RNA extraction.

C. difficile cultivation and quantification  Differential plating was performed to quantify both C. difficile spores and vegetative cells respectively.  Cecal samples were weighed and serially diluted under anaerobic conditions with anaerobic PBS.  Samples were plated on CCFAE plates (fructose agar plus cycloserine, cefoxitin, and erythromycin) and counted for CFU after 24 hours of anaerobic growth at 37 C (Buggy).  It is important to note that the germination agent taurocholate was omitted from these plates in order to only quantify vegetative cells.  Undiluted samples with then heated at 60 C for 30 minutes in order to eliminate vegetative cells and leave only spores (Sorg).  These samples were then serially diluted under anaerobic conditions in anaerobic PBS and plated on TCCFAE (CCFAE plus taurocholate).  Plating was simultaneously done for heated samples on CCFAE to ensure all vegetative cells had been eliminated.  After 24 hours of anaerobic incubation at 37 C CFUs from spores were quantified.  

C. difficile Toxin Assay  Vero cell rounding assay was performed on mouse cecal content as previously described (Leslie, 2014).

DNA extraction, library preparation, and sequencing  Total bacterial DNA was extracted from a cecal sample per mouse following necropsy using the MOBIO PowerSoil-htp 96-well soil DNA isolation kit.  Amplicon libraries of the V4 region within the 16S rRNA gene were generated and sequenced using an Illumina MiSeq as previously described (Kozich).  Metagenomic libraries were created from extracted, pooled genomic DNA with treatment group using a customized Nextera XT genomic library construction protocol (Illumina) as described in (Kozich).  Sequencing for these libraries was performed on an Illumina HiSeq 2500 with a 2x250 bp read length on the High Output setting.

RNA extraction, library preparation, and sequencing  This procedure was adapted from a protocol that can be found here ().  3 ml of lysis buffer (2% SDS, 16 mM EDTA and 200 mM NaCl) contained in a 50 ml polypropylene conical tube was first heated for 5 minutes in a  boiling water bath.  The hot lysis buffer was added to the frozen and ground cecal content.  The mixture was boiled with periodic vortexing for another 5 minutes.  After boiling, an equal volume of 37° C acid phenol/chloroform was added to the cecal content lysate and incubated at 37° C for 10 minutes with periodic vortexing.  The mixture was the centrifuged at 2,500 g at 4° C for 15 minutes.  
	The aqueous phase was then transferred to a sterile tube and an equal volume of acid phenol/chloroform was added.  This mixture was vortexed and centrifuged at 2,500 g at 4° for 5 minutes.  The process was repeated until aqueous phase was clear.  The last extraction was performed with chloroform/isoamyl alcohol to remove acid phenol.  An equal volume of isopropanol was added and the extracted nucleic acid was incubated overnight at -20° C.  The following day the sample was centrifuged at 12000 g at 4° C for 45 minutes.  The pellet was washed with 0° C 100% ethanol and resuspended in 200 μl of RNase-free water.  
	Following the manufacterer's protocol, samples were then treated with 2 μl of Turbo Dnase for 30 minutes.  RNA samples were retrieved using the Zymo Quick-RNA MiniPrep according the manufacterer's protocol.  Completion of the reaction was assessed using PCR for the V6 region of the 16S rRNA gene (Kozich).  Quality and integrity of RNA was assessed using the  Agilent RNA 6000 Nano kit for Total Prokaryotic RNA.  The Ribo-Zero Gold rRNA Removal Kit (Epidemiology) was then used to deplete Prokaryotic and Eukaryotic rRNA from the samples according the manufacterer's protocol.  
	Prior to library construction, quality and integrity as measured again using the Agilent RNA 6000 Pico Kit and RNA-Seq libraries were made constructed using the TruSeq RNA Library Preparation Kit v2.  2x50 bp sequencing was finally performed using an Illumina HiSeq 2500.  A more detailed protocol for this entire procedure can be found in Supplementary Material.

Sequence curation  16S rRNA gene amplicon sequences were curated using mothur (v.1.37) as previously described (29, 30). Very briefly, sequences were binned into operational taxonomic units (OTUs) using a 3% dissimilarity cutoff.  Taxonomic assignments were determined by using a naive Bayesian classifier with the Ribosomal Database Project (RDP) training set (version 10)requiring an 80% bootstrap confidence score (31). In parallel a mock community of known 16S rRNA gene regions was sequenced. Analysis of the this data indicated that the error rate following our curation procedure was >0.02%.
	Curation of metagenomic and metatranscriptomic sequencing reads was performed in a two step process.  Residual forard and reverse sequenceing primer sequence was trimmed using CutAdapt and the correct primer sequences corresponding to each library.  Next, reads were quality trimmed based on Q-scores using Sickle, with the additional settings ___ (ref).
	Metatranscriptomic reads were further curated by screening out contaminating sequence including remaining reads from bacterial rRNA, mouse (mus musculus) exome, and C. difficile str. 630 (when necessary).  This was accomplished using Bowtie2 short read aligner using the 1 allowed mismatch setting and the corresponding reference (accession numbers for references).

Metagenomic Assembly, CONCOCT, gene calling, and annotation  Reads were assembled using Megahit with the recommended settings (ref).  Short contigs were removed (<250 bp) and potential genes were called using Prodigal on the default settings (ref) and short genes were removed from the dataset (<250 bp or <80 aa).  
Concoct to retrieve separate genomes

Read Mapping and normalization  Read mapping for downstream analysis was performed using Bowtie2 and the default stringent settings (0 mismatches).  The resulting SAM files were then screened for duplicates due to PCR using the markDuplicates function within Picard.  The remaining mappings were converted to idxstats format using Samtools and the read counts per gene were tabulated.  Read counts were pooled with discordant pair mappings and counts were then normalized to read length and gene length.  Normalized metatranscriptomic read counts underwent an additional step by normalizing to metagenomic coverage (DNA read count) on a per gene basis.  All metagenomic and metatranscriptomic datasets were subsampled respectively to the same number of reads as 90% of the total reads in the sample with the fewest reads of the given sequencing type (DNA or RNA).

Statistical analysis



Metabolic modeling  Single organism level metabolic modeling was accomplished using KEGG gene annotations.  



$$log_{2}\Big(\frac{\sum_{} T_i}{e_{o}} - \frac{\sum_{} T_o}{e_{i}} \Big)$$



Protocols, code, and are available in a Github repository 
https://github.com/SchlossLab/Jenior_Transcriptomics_2015


###References

[1]
[2]
[3]
[4]
[5]
[6]
[7]
[8]
[9]
[10]



