---
title: "*Clostridium difficile* colonizes alternative nutrient niches during infection across distinct murine gut communities"
csl: mbio.csl
output:
 word_document:
  keep_md: true
  reference_docx: manuscript_format.docx
bibliography: references.bib
---

**Authors:** Matthew L. Jenior, Jhansi L. Leslie, Vincent B. Young, and Patrick D. Schloss^\*^

[Comment:
Lack of consistency across manuscript on...
* Whether gf mice are "germ free", "ex-germ free", "gnotobiotic", etc.
* How antibiotic-treated mice are named
* Whether to refer to C. difficile as "CDI", "630", etc.
]

```{r startup, echo=F, message=F, warning=F, cache=F}

# Load dependencies
deps <- c('shape', 'vegan', 'klaR', 'igraph', 'matrixStats', 'flux', 'knitr','plyr')
for (dep in deps){
 if (dep %in% installed.packages()[,"Package"] == FALSE){
  install.packages(as.character(dep), quiet=TRUE);
 }
 library(dep, verbose=FALSE, character.only=TRUE)
}

# Define function for formatting growth curves for statistical analysis
format_curve <- function(raw_exp_data, exp_group, raw_control_data){
 formatted_data <- c()
 control_data <- c()
 for (time in 1:nrow(raw_exp_data)){
  temp_exp <- cbind(exp_group, time, raw_exp_data[time,])
  formatted_data <- rbind(formatted_data, temp_exp)
  temp_control <- cbind('control', time, raw_control_data[time,])
  control_data <- rbind(control_data, temp_control)
 }
 formatted_data <- as.data.frame(rbind(control_data, formatted_data))
 colnames(formatted_data) <- c('substrate','time','od')
 formatted_data$od <- as.numeric(as.character(formatted_data$od))

 return(formatted_data)
}

# Get data files
# Mouse metadata
metadata <- read.delim('data/metadata.tsv', sep='\t', header=T, row.names=1)
# Figure 1 data
cfu <- read.delim('data/wetlab_assays/cfu.dat', sep='\t', header=T)
toxin <- read.delim('data/wetlab_assays/toxin_titer.dat', sep='\t', header=T)
# Figure 4 data
network <- read.delim('data/metabolic_models/cefoperazone_630.bipartite.files/graph.tsv', header=FALSE, sep='\t')
# Figure 5 data
growth <- read.delim('data/wetlab_assays/cd630_growth.tsv', sep='\t', header=TRUE, row.names=1)

# Set a seed to the answer to the ultimate question of life, the universe, and everything
set.seed(42)

```


### Abstract

*Clostridium difficile* infections (CDIs) have become the largest single cause of hospital-acquired infection in the United States. CDI susceptibility is most frequently associated with previous antibiotic exposure, which disrupts the gut bacterial community. This has been described for multiple antibiotic classes, which result in distinct, gut communities, each presenting separate metabolic challenges to C. difficile. We hypothesized that *C. difficile* adapts its physiology to nutrient availability within differentially susceptible gut environments. Utilizing an in vivo model of CDI, we demonstrated that *C. difficile* highly colonized the cecum of mice receiving one of three individual antibiotic pretreatments. We found levels of spore and toxin production varied between each group, both processes partially regulated by environmental nutrient concentrations. To more closely investigate metabolic responses of *C. difficile* during infection, we performed transcriptomic analysis of the pathogen from cecal content of infected mice. This revealed expression variation in numerous catabolic pathways for various carbon sources. To assess which substrates *C. difficile* was exploiting, we developed a transcriptomic-enabled genome-scale metabolic model of *C. difficile* and a metabolite scoring algorithm that leveraged network architecture. With this platform, we identified carbon sources used by *C. difficile* asymmetrically between infection models. These results were validated through correlation with untargeted mass spectrometry analysis from each condition. Our results supported the hypothesis that *C. difficile* indeed metabolized alternative carbon sources across colonized environments. These data also highlighted conserved elements of C. difficile's metabolic strategy, specifically consumption of host-derived aminoglycans and Stickland fermentation substrates.


### Importance

Individuals who are treated with antibiotics are at significant risk for a *C. difficile* infection. An infected patient will be treated with additional antibiotics and may experience a recurrent infection that can only be effectively treated via a fecal microbiota transplant. *C. difficile*'s ability to colonize patients treated with diverse antibiotics suggests that the pathogen is effective at adapting its metabolism to exploit the niche space that is opened by antibiotic therapies. Here, we demonstrate that *C. difficile* does exploit unique nutrient niches across environments that were perturbed with different antibiotics. Our analysis provides evidence that *C. difficile* virulence may be driven by accessibility of specific carbon sources during infection. By integrating the ecology of the host microbiota with the metabolic plasticity of *C. difficile* it will be possible to develop specific therapies that limit *C. difficile*'s access to novel niche space and prevent infection.


### Introduction

Infection by the Gram-positive, spore-forming bacterium *Clostridium difficile* has increased in both prevalence and severity across numerous countries during the last decade [@Lessa2012]. In the United States, *C. difficile* was estimated to have caused more than 500,000 infections and resulted in approximately $4.8 billion worth of acute care costs in 2014 [@Lessa2015]. *C. difficile* infection (CDI) causes an array of toxin-mediated symptoms ranging from abdominal pain and diarrhea to more life-threatening conditions including pseudomembraneous colitis and toxin megacolon. Prior treatment with antibiotics, which disrupts the gut microbiota, is the most common risk factor associated with susceptibility to CDI [@Leffler2015; @Britton2014].

*C. difficile* is unable to colonize a conventionally reared mouse. However, perturbation of the normal murine microbiota by numerous antibiotics has been shown to induce susceptibility to colonization by *C. difficile* [@Chen2008;@Theriot2014;@Schubert2015]. Previous studies have shwon that diverse antibiotics can have a unique impact on the structure of the gut microbiota leading to communities that were highly susceptible to *C. difficile* colonization. In addition, antibiotic treatments also altered the gut metabolome, which increased the concentrations of some *C. difficile* growth substrates [@Antunes2011; @Theriot2014; @Ferreyra2014; @Jump2014]. Taken together these results are a strong indication that the healthy gut microbiota inhibits the growth of *C. difficile* by limiting the availability of substrates that it needs to grow. The ability of an intact gut community to exclude *C. difficile* colonization is suggestive of the nutrient-niche hypothesis in which an organism must be able to utilize a subset of available resources better than all competitors to colonize the intestine [@Freter1983; @Wilson1988].

Based on genomic and *in vitro* growth characteristics, *C. difficile* appears able to fill multiple nutrient niches. *C. difficile* has a relatively large and mosaic genome, it is amenable [Comment: I’m not sure what this means here. Amenable means that it is open to suggestion. I think you means that it can grow on a variety of substrates.] to a variety of growth substrates, and is able to colonize a diverse array of hosts suggesting that that it is an ecological generalist [@Sebaihia2006;@Tracy2012;@Songer2006]. Several forms of *C. difficile* carbon metabolism have also been shown to regulate pathogenesis. *In vitro* transcriptomic analyses suggests that high concentrations of easily metabolized carbon sources, such as glucose or amino acids, inhibit toxin gene expression and sporulation [@Neumann-Schaal2015; @Nawrocki2016]. Transcriptomic studies of *C. difficile* have shown that... [Comment: fill in the blank with a quick summary of their findings]. These transcriptomic studies have mainly focused on *in vitro* growth [@Matamouros2007; @Antunes2011b] or colonization of germfree mice [@Janoir2013; @Kansau2016]. Although these analyses are informative, they are either primarily directed toward the expression of virulence factors or lack the context of the gut microbiota which *C. difficile* must compete against for substrates. Metabolomic analyses have also been used to assay changes in bacterial metabolism as they relate to CDI and have characterized the levels of germinants and sporulation-inducing molecules [@Theriot2014;@Jump2014]; however, metabolomic approaches are unable to a attribute a metabolite to a specific organisms in the milieu. Thus metabolomics more closely represents the echoes of total community metabolism, not the currently active processes of any one population. It has thus far not been possible to study *C. difficile*’s metabolism *in vivo*.

~~Genome-scale metabolic models have proven useful for accurately determining a given organism's nutritional requirements in different scenarios [@Patil2005]. Integrating transcriptomic data with genome-scale metabolic modeling has previously aided in identifying the most active aspects of an organism’s metabolism and which substrates are preferred by the organism [@Patil2005;@Borenstein2008;@Suthers2009]. Applying these methods to study *C. difficile* colonization would allow us to directly test the nutrient-niche hypothesis.~~ [Comment: This paragraph can go away. It kind of sticks out here. The next sentence picks up well with the end of the previous paragraph.]

To overcome these limitations, we used *in vivo* transcriptomic and metabolomic analysis of *C. difficile* in specific pathogen free (SPF) mice to better understand the active metabolic pathways in a more realistic model of infection. Based on the ability of *C. difficile* to grow on a diverse array of carbon sources and its ability to colonize a variety of communities, we hypothesized that *C. difficile* adapts its metabolism to fit the context of the community it is attempting colonize. To test this hypothesis, we employed a mouse model of infection to compare the response of *C. difficile* to the gut environment caused by three distinct classes of antibiotics. By characterizing the transcriptome of *C. difficile* in these different communities and the metabolome of the communities, we were able to generate a systems model to directly test the nutrient-niche hypothesis.


~~Those chosen included streptomycin, cefoperazone, and clindamycin (Table 1) since each significantly impact the structure and diversity of the cecal microbiome uniquely (Fig. S2A and S2B). It has been further demonstrated that at 18 hours after being introduced to a cefoperazone-pretreated mouse, *C. difficile* reached its maximum vegetative cell density in the cecum with limited sporulation [@Koenigsknecht2015]. This provided a single timepoint to measure the largest population of metabolically active *C. difficile*. We identified differences in the amount of sporulation and toxin activity between each of the conditions tested, further supporting our prediction on environmentally driven changes to metabolism. This was further supported by large-scale differences in expression of *C. difficile* metabolic pathways between each infection model. We then predicted that these distinct adaptive responses were necessitated because *C. difficile* had encountered unique subsets of nutrients in each environment. Integration of transcriptomic data with genome-scale metabolic modeling revealed that *C. difficile* utilized alternative carbon sources across conditions, adapting it's metabolic strategy in each instance. We then validated modeling-based conclusions with untargeted metabolomic analysis to most accurately capture the changes in *C. difficile* behavior. This multi-omic analysis demonstrated that in each antibiotic-pretreatment model, as well as in monoassociated germfree mice, *C. difficile* adapted its nutrient utilization profile to most effectively colonize. Our data also suggests that in addition to this metabolic plasticity, *C. difficile* also maintained at least some focus on catabolism of amino acids and host-derived aminoglycans in each infected condition. These findings are important as they have implications for the future design of measures for the prevention or elimination of *C. difficile* colonization from the gut.~~ [Comment: This is too resulty. It can be cut.]


###Results

```{r figure_1, echo=F, warning=F, message=F, results='hide', cache=F}

cfu <- subset(cfu, cage < 4 ) # Remove uninfected controls
cfu_raw <- cfu
toxin_raw <- toxin

# Format CFU data and collect summary statistics
cfu[cfu == 0] <- 100
cfu$cfu_vegetative <- log10(cfu$cfu_vegetative)
cfu$cfu_spore <- log10(cfu$cfu_spore)
cfu$mouse <- NULL
cfu$cage <- NULL
cfu$treatment <- factor(cfu$treatment, levels=c('streptomycin', 'cefoperazone', 'clindamycin', 'germfree', 'conventional'))
vegetative_cfu <- cfu
vegetative_cfu$cfu_spore <- NULL
spore_cfu <- cfu
spore_cfu$cfu_vegetative <- NULL
cef <- as.numeric(median(vegetative_cfu[vegetative_cfu$treatment == 'cefoperazone', 2]))
strep <- as.numeric(median(vegetative_cfu[vegetative_cfu$treatment == 'streptomycin', 2]))
clinda <- as.numeric(median(vegetative_cfu[vegetative_cfu$treatment == 'clindamycin', 2]))
gf <- as.numeric(median(vegetative_cfu[vegetative_cfu$treatment == 'germfree', 2]))
conv <- as.numeric(median(vegetative_cfu[vegetative_cfu$treatment == 'conventional', 2]))
vege_medians <- c(strep, cef, clinda, gf, conv)
vege_medians[vege_medians == 2.0] <- 1.6
cef <- as.numeric(median(spore_cfu[spore_cfu$treatment == 'cefoperazone', 2]))
strep <- as.numeric(median(spore_cfu[spore_cfu$treatment == 'streptomycin', 2]))
clinda <- as.numeric(median(spore_cfu[spore_cfu$treatment == 'clindamycin', 2]))
gf <- as.numeric(median(spore_cfu[spore_cfu$treatment == 'germfree', 2]))
conv <- as.numeric(median(spore_cfu[spore_cfu$treatment == 'conventional', 2]))
spore_medians <- c(strep, cef, clinda, gf, conv)
spore_medians[spore_medians == 2.0] <- 1.6
vegetative_cfu$color <- ifelse(vegetative_cfu$cfu_vegetative == 2.0, 'gray50', 'black')
vegetative_cfu$cfu_vegetative[vegetative_cfu$cfu_vegetative == 2.0] <- 1.6
spore_cfu$cfu_spore[spore_cfu$cfu_spore == 2.0] <- 1.6

# Format toxin data and find summary statistics
toxin$mouse <- NULL
toxin$cage <- NULL
toxin$treatment <- factor(toxin$treatment, levels=c('Streptomycin', 'Cefoperazone', 'Clindamycin', 'Germfree', 'Conventional'))
cef <- as.numeric(median(toxin[toxin$treatment == 'Cefoperazone', 2]))
strep <- as.numeric(median(toxin[toxin$treatment == 'Streptomycin', 2]))
clinda <- as.numeric(median(toxin[toxin$treatment == 'Clindamycin', 2]))
gf <- as.numeric(median(toxin[toxin$treatment == 'Germfree', 2]))
conv <- as.numeric(median(toxin[toxin$treatment == 'Conventional', 2]))
toxin_medians <- c(strep, cef, clinda, gf, conv)
toxin_medians[toxin_medians <= 2.0] <- 1.9
toxin$titer[toxin$titer <= 2.0] <- 1.9

#----------------------------#

# Calculate significant differences and correct p values

# Toxin
tox_p_values <- round(p.adjust(c(wilcox.test(subset(toxin_raw, treatment == 'Germfree')$titer, subset(toxin_raw, treatment == 'Cefoperazone')$titer, exact=F)$p.value,
                 wilcox.test(subset(toxin_raw, treatment == 'Germfree')$titer, subset(toxin_raw, treatment == 'Clindamycin')$titer, exact=F)$p.value,
                 wilcox.test(subset(toxin_raw, treatment == 'Germfree')$titer, subset(toxin_raw, treatment == 'Streptomycin')$titer, exact=F)$p.value,
                 wilcox.test(subset(toxin_raw, treatment == 'Cefoperazone')$titer, subset(toxin_raw, treatment == 'Clindamycin')$titer, exact=F)$p.value,
                 wilcox.test(subset(toxin_raw, treatment == 'Cefoperazone')$titer, subset(toxin_raw, treatment == 'Streptomycin')$titer, exact=F)$p.value,
                 wilcox.test(subset(toxin_raw, treatment == 'Clindamycin')$titer, subset(toxin_raw, treatment == 'Streptomycin')$titer, exact=F)$p.value), method='holm'), digits=3)
gf_vs_cef_tox_p <- tox_p_values[1]
gf_vs_clinda_tox_p <- tox_p_values[2]
gf_vs_strep_tox_p <- tox_p_values[3]
cef_vs_clinda_tox_p <- tox_p_values[4]
cef_vs_strep_tox_p <- tox_p_values[5]
clinda_vs_strep_tox_p <- tox_p_values[6]

# Spores
spore_p_values <- round(p.adjust(c(wilcox.test(subset(cfu_raw, treatment == 'germfree')$cfu_spore, subset(cfu_raw, treatment == 'cefoperazone')$cfu_spore, exact=F)$p.value,
                  wilcox.test(subset(cfu_raw, treatment == 'germfree')$cfu_spore, subset(cfu_raw, treatment == 'clindamycin')$cfu_spore, exact=F)$p.value,
                  wilcox.test(subset(cfu_raw, treatment == 'germfree')$cfu_spore, subset(cfu_raw, treatment == 'streptomycin')$cfu_spore, exact=F)$p.value,
                  wilcox.test(subset(cfu_raw, treatment == 'clindamycin')$cfu_spore, subset(cfu_raw, treatment == 'cefoperazone')$cfu_spore, exact=F)$p.value,
                  wilcox.test(subset(cfu_raw, treatment == 'streptomycin')$cfu_spore, subset(cfu_raw, treatment == 'cefoperazone')$cfu_spore, exact=F)$p.value,
                  wilcox.test(subset(cfu_raw, treatment == 'streptomycin')$cfu_spore, subset(cfu_raw, treatment == 'clindamycin')$cfu_spore, exact=F)$p.value), method='holm'), digits=3)
gf_vs_cef_spore_p <- spore_p_values[1]
gf_vs_clinda_spore_p <- spore_p_values[2]
gf_vs_strep_spore_p <- spore_p_values[3]
clinda_vs_cef_spore_p <- spore_p_values[4]
strep_vs_cef_spore_p <- spore_p_values[5]
strep_vs_clinda_spore_p <- spore_p_values[6]

gf_both_p <- max(c(gf_vs_cef_tox_p,gf_vs_clinda_tox_p,gf_vs_strep_tox_p,gf_vs_cef_spore_p,gf_vs_clinda_spore_p,gf_vs_strep_spore_p))


```


**Levels of *C. difficile* sporulation and toxin activity vary among different communities.**
Conventionally-reared specific pathogen free mice were treated with one of three different antibiotics. These included streptomycin, cefoperazone, and clindamycin (Table 1) (Fig. S1), which were selected because they each significantly impact the structure and diversity of the cecal microbiome uniquely (Fig. S2A and S2B). In addition, we mono-associated germ free mice with *C. difficile* to understand its physiology in the absence of other microbiota. We  measured sporulation and toxin production at 18 hours post infection because that corresponds to the time when C. difficile reaches its maximum vegetative cell density in the cecum with limited sporulation [@Koenigsknecht2015]. There was not a significant difference in the number of vegetative cells between any susceptible condition tested (Fig. 1A). All antibiotic treated (Table 1) and germfree (GF) mice were colonized to ~1×10^8^ colony forming units (cfu) per gram of cecal content, while untreated mice maintained colonization resistance to *C. difficile*. Despite having the same number of vegetative *C. difficile* cells, large more spores were detected in ex-GF mice than in the antibiotic treated mice (*P* = `r gf_vs_strep_spore_p`, `r gf_vs_cef_spore_p`, and `r gf_vs_clinda_spore_p`; Fig. 1B). There was significantly more toxin activity in ex-GF animals than any other colonized group (all *P* < `r max(c(gf_vs_strep_tox_p, gf_vs_cef_tox_p, gf_vs_clinda_tox_p))`)~~, but toxin titer also varied between antibiotic treatment groups~~[Comment: no it doesn't. There's one mouse in the strep and cef groups that has toxin. Also, is the median 2.3 or below the limit of detection? It looks like it should be below the LOD.] (Fig. 1C). ~~Although similar toxin activity was found in both the cefoperazone and clindamycin-treated groups, toxin titer was below the limit of detection in most streptomycin-treated animals.~~ [Comment: not really. there's a single animal that sticks out for strep and cef. For 1C, the y-axis can be linear - 10^2.3 is 200 and 10^3 is 1000]. These results indicated that *C. difficile* colonized different communities to a consistently high level and that the antibiotic perturbed communities had similar levels of spores and toxin. To investigate the physiology of *C. difficile* when colonizing distinct susceptible gut environments, we performed whole transcriptome analysis of *C. difficile* growing in the cecum during infection.



** *C. difficile* alters its gene expression pathways when colonizing different communities.**
We first attempted to measure differential expression of specific genes associated with *in vivo* phenotype changes reported in previous studies. These included genes involved in sporulation, the pathogenecity locus, quorum sensing, and metabolite-regulated sigma factors (Fig. S3); however no clear trends between gene expression and colonization, sporulation, or toxin production were evident. It became evident that colonization of each treatment group resulted in different levels of expression for each group of genes. This further indicated that *C. difficile* may adapt its metabolism to the environment it is colonizing.

We next focused on specific gene families known to contribute to *C. difficile* metabolism (Table S1 [Comment: Table S1 needs to also include the %s to help direct the reader to Figure 2] and Fig. 2A). Genes involved in amino acid catabolism, including those that encoded enzymes involved in Stickland fermentation and general peptidases, had the highest level of expression. This indicated that catabolizing environmental amino acids were generally important for the growth of *C. difficile* during infection, regardless of the structure of the surrounding community (Fig. 2B). Although there were genes that were equally-well expressed regardless of the structure of the surrounding community, there were specific gene families and specific genes that were unique to each community. Mice treated with cefoperazone tended to have more expression of genes in the ABC Sugar Transporter and Sugar Alcohol Catabolism (e.g. mannitol) families and fewer genes in the PTS Transporter family than the other treatment groups. Mice treated with clindamycin tended to have more expression of genes from the Disaccharide Catabolism (e.g. alpha and beta-galactosidase), Fermentation Product Synthesis (e.g. butyrate and butanol), and PTS Transporter families. Genes from the Sugar Alcohol Catabolism and ABC Sugar Transporter families were not highly expressed in clindamycin-treated mice. Finally, mice treated with streptomycin tended to have higher levels of expression of genes from the Sugar Alcohol Catabolism (e.g. sorbitol) and PTS Transporter families. Combined, these results suggested that catabolism of amino acids and specific carbohydrates are core components of the *C. difficile* nutritional strategy during infection. These results also supported the hypothesis that *C. difficile* adapted its metabolism across different susceptible environments.


```{r figure_3, echo=F, warning=F, message=F, results='hide', cache=F}

# Format directed graph
raw_graph <- graph.data.frame(network, directed=TRUE)

# Decompose graph
decomp_whole_graph <- decompose.graph(raw_graph)

# Get largest component and get node information
largest_component <- which.max(sapply(decomp_whole_graph, vcount))
largest_whole_graph <- decomp_whole_graph[[largest_component]]
print(length(as.vector(grep('K', V(largest_whole_graph)$name, value=TRUE)))) # 404
print(length(as.vector(grep('C', V(largest_whole_graph)$name, value=TRUE)))) # 666

# Find strongly-connected components
largest_scc <- rownames(as.data.frame(clusters(largest_whole_graph, mode='strong')[1]))

#----------------------------#

# Determine some statistics about graph

# Print a summary of nodes and edges for entire graph
summary(raw_graph)
print(length(as.vector(grep('K', V(raw_graph)$name, value=TRUE))))
print(length(as.vector(grep('C', V(raw_graph)$name, value=TRUE))))

# Find degrees of nodes
graph_indegree <- as.data.frame(degree(raw_graph, v=V(raw_graph), mode='in'))
graph_outdegree <- as.data.frame(degree(raw_graph, v=V(raw_graph), mode='out'))
graph_undirected <- as.data.frame(degree(raw_graph, v=V(raw_graph), mode='all'))

# Calculate betweensness of entrire graph
graph_betweenness <- as.data.frame(betweenness(raw_graph, normalized=TRUE))

# Calculate closeness of entrire graph
graph_closeness_in <- as.data.frame(closeness(raw_graph, vids=V(raw_graph), mode='in', normalized=TRUE))
graph_closeness_out <- as.data.frame(closeness(raw_graph, vids=V(raw_graph), mode='out', normalized=TRUE))
graph_closeness_total <- as.data.frame(closeness(raw_graph, vids=V(raw_graph), mode='total', normalized=TRUE))

```


**Genome-scale metabolic model structure underscores known *C. difficile* biology.**
To further support the hypothesis that *C. difficile* modifies its metabolism in different communities, we developed a metabolic network of the *C. difficile* genome. A metabolic model would allow us to map transcript counts to predict the flow of metabolites through *C. difficile*. We represented the metabolic network of *C. difficile* as a directed bipartite graph using the genome and biochemical reaction annotations available in the KEGG database [@Ogata1999][Comment: Confirm this is the correct citation]. Enzymes and metabolites were represented by nodes, and their interactions by directed connecting edges (Fig. 3A). The complete network of *C. difficile* str. 630 contained 447 [Comment: need R code to generate 447] enzymes and 758 [Comment: need R code to generate 758] metabolites, with 2135 [Comment: need R code] directed edges. To validate our metabolic network, we analyzed network topology by calculating two metrics of centrality, betweenness centrality (BC) and closeness centrality (CC), for each node. BC is the quantity of shortest paths connecting all other member nodes of a network that pass through a given node and quantifies the amount of influence a given hub has on the overall flow of metabolism through the network [@Potapov2005]. CC is the reciprocal sum of the lengths of those shortest paths quantified by the BC and measures how essential a given node is the the overall structure of the metabolic network [@Koschutzki2008]. Calculating CC and BC allowed us to identify the nodes that were critical to the structure of the metabolic network and determine whether these patterns reflected known patterns in *C. difficile* and general bacterial metabolism (Table S2). We combined both centrality measures to rank both the enzymes and their inferred metabolites by their importance. Among the 10 most central enzymes, 5 were critical to glycolysis, the pentose phosphate pathway, and the citric acid cycle (2-dehydro-3-deoxyphosphogluconate aldolase, aspartate aminotransferase, pyruvate-flavodoxin oxidoreductase, formate C-acetyltransferase, and 1-deoxy-D-xylulose-5-phosphate synthase). The majority of the other high ranking enzymes were also components of these pathways and for the metabolism of amino acids (Table S2). Similarly, the high ranking metabolites identified 6 metabolites as central nodes to the metabolism of *C. difficile* (pyruvate, acetyl-CoA, 2-oxoglutarate, D-4-hydroxy-2-oxoglutarate, D-glyceraldehyde 3-phosphate, and L-glutamate). These metabolites are intermediaries of glycolysis and the citric acid cycle and contribute to numerous intracellular pathways (pyruvate, acetyl-CoA, and L-glutamate). Notably absent from the most well-connected metabolites were molecules like ATP or NADH. Their exclusion is likely a byproduct of the KEGG LIGAND reference used for network construction, which excludes cofactors from most biochemical reactions. These results collectively support the validity of our gene-based metabolic model of *C. difficile* str. 630.


```{r figure_4, echo=F, warning=F, message=F, results='hide', cache=F}

growth <- as.data.frame(t(growth))

# Seperate to groups of each growth substrate and format
sorbitol <- cbind(growth$sorbitol_1, growth$sorbitol_2, growth$sorbitol_3) - growth$sorbitol_blank
sorbitol[sorbitol < 0] <- 0
mannitol <- cbind(growth$mannitol_1, growth$mannitol_2, growth$mannitol_3) - growth$mannitol_blank
mannitol[mannitol < 0] <- 0
salicin <- cbind(growth$salicin_1, growth$salicin_2, growth$salicin_3) - growth$salicin_blank
salicin[salicin < 0] <- 0
acetylglucosamine <- cbind(growth$acetylglucosamine_1, growth$acetylglucosamine_2, growth$acetylglucosamine_3) - growth$acetylglucosamine_blank
acetylglucosamine[acetylglucosamine < 0] <- 0
acetylneuraminate <- cbind(growth$acetylneuraminate_1, growth$acetylneuraminate_2, growth$acetylneuraminate_3) - growth$acetylneuraminate_blank
acetylneuraminate[acetylneuraminate < 0] <- 0
no_carb <- cbind(growth$noCarb_1, growth$noCarb_2, growth$noCarb_3) - growth$noCarb_blank
no_carb[no_carb < 0] <- 0
no_aa <- cbind(growth$noAA_1, growth$noAA_2, growth$noAA_3) - growth$noAA_blank
no_aa[no_aa < 0] <- 0


# Prepare data for statistical tests
sorbitol_test <- format_curve(sorbitol, 'sorbitol', no_carb)
mannitol_test <- format_curve(mannitol, 'mannitol', no_carb)
salicin_test <- format_curve(salicin, 'salicin', no_carb)
acetylglucosamine_test <- format_curve(acetylglucosamine, 'acetylglucosamine', no_carb)
acetylneuraminate_test <- format_curve(acetylneuraminate, 'acetylneuraminate', no_carb)
no_aa_test <- format_curve(no_aa, 'no_amino_acids', no_carb)

# Calculate differences and correct p values
sorbitol_p <- summary(aov(formula=od ~ substrate * time, data=sorbitol_test))[[1]][["Pr(>F)"]][3]
mannitol_p <- summary(aov(formula=od ~ substrate * time, data=mannitol_test))[[1]][["Pr(>F)"]][3]
salicin_p <- summary(aov(formula=od ~ substrate * time, data=salicin_test))[[1]][["Pr(>F)"]][3]
acetylglucosamine_p <- summary(aov(formula=od ~ substrate * time, data=acetylglucosamine_test))[[1]][["Pr(>F)"]][3]
acetylneuraminate_p <- summary(aov(formula=od ~ substrate * time, data=acetylneuraminate_test))[[1]][["Pr(>F)"]][3]
no_aa_p <- summary(aov(formula=od ~ substrate * time, data=no_aa_test))[[1]][["Pr(>F)"]][3]
aov_p_values <- c(sorbitol_p,mannitol_p,salicin_p,acetylglucosamine_p,acetylneuraminate_p,no_aa_p)
aov_p_values <- p.adjust(aov_p_values, method='holm')
sorbitol_cp <- aov_p_values[1]
mannitol_cp <- aov_p_values[3]
salicin_cp <- aov_p_values[4]
acetylglucosamine_cp <- aov_p_values[6]
acetylneuraminate_cp <- aov_p_values[7]
experimental_cp <- c(sorbitol_cp,mannitol_cp,salicin_cp,acetylglucosamine_cp,acetylneuraminate_cp)
no_aa_cp <- aov_p_values[9]

# Find medians
sorbitol_median <- apply(sorbitol, 1, median)
mannitol_median <- apply(mannitol, 1, median)
salicin_median <- apply(salicin, 1, median)
acetylglucosamine_median <- apply(acetylglucosamine, 1, median)
acetylneuraminate_median <- apply(acetylneuraminate, 1, median)
no_carb_median <- apply(no_carb, 1, median)
no_aa_median <- apply(no_aa, 1, median)

#----------------------------#

# Analyze growth curves

# Maximum positive growth rate
sorbitol_max_rate <- round(diff(sorbitol_median)[which.max(diff(sorbitol_median))], digits=3)
acetylglucosamine_max_rate <- round(diff(acetylglucosamine_median)[which.max(diff(acetylglucosamine_median))], digits=3)
acetylneuraminate_max_rate <- round(diff(acetylneuraminate_median)[which.max(diff(acetylneuraminate_median))], digits=3)
mannitol_max_rate <- round(diff(mannitol_median)[which.max(diff(mannitol_median))], digits=3)
salicin_max_rate <- round(diff(salicin_median)[which.max(diff(salicin_median))], digits=3)
no_aa_max_rate <- round(diff(no_aa_median)[which.max(diff(no_aa_median))], digits=3)
no_carb_max_rate <- round(diff(no_carb_median)[which.max(diff(no_carb_median))], digits=3)

# Maximum OD
sorbitol_max_od <- round(max(sorbitol_median), digits=3)
acetylglucosamine_max_od <- round(max(acetylglucosamine_median), digits=3)
acetylneuraminate_max_od <- round(max(acetylneuraminate_median), digits=3)
mannitol_max_od <- round(max(mannitol_median), digits=3)
salicin_max_od <- round(max(salicin_median), digits=3)
no_aa_max_od <- round(max(no_aa_median), digits=3)
no_carb_max_od <- round(max(no_carb_median), digits=3)

```

**Metabolite importance algorithm reveals adaptive nutritional strategies of *C. difficile* during infection of distinct environments.** Having constructed and validated the metabolic model for *C. difficile*, we mapped transcriptomic data to the network infer which metabolites *C. difficile* most likely obtained from its environment. The importance of each metabolite was measured as the log~2~-transformed difference between the average transcript level for each enzyme that used the metabolite as a substrate and those that generated it as a product (Fig. 3B). If the expression of genes for enzymes that produce the metabolite were low, then we reasoned that a metabolite with a high importance score was most likely obtained from the environment. We adopted a Monte Carlo-style simulation to test whether the observed importance score was significantly different from what we would expect if the associations were generated randomLy [@Bonett2002] (Fig. 3C). We excluded all metabolites from our downstream analyses if they fell within the 95% confidence interval for a random association. This resulted in the identification of XXX, XXX, and XXX metabolites that had significant importance scores in the cefoperazone, streptomycin, and clindamycin treatement groups, respectively (Table S3).

To identify the core metabolites that were most essential for *C. difficile* growth, regardless of the environment, we identified 50 highest scoring metabolites from each treatment group and recalculated the median importance across all conditions (Fig. 4A) [Comment: It is not clear why you would need to recalculate the scores nor why you would]. The host-derived aminoglycan N-acetyl-D-glucosamine was found to the have the highest median importance of all shared metabolites, which has been shown to be a readily available source of carbon and nitrogen utilized by *C. difficile* [reference]. Components of the Stickland fermentation pathway including proline, 3−hydroxybutanoyl−CoA, and formate also had high importance scores across all conditions [@Aboulnaga2013;@Fonknechten2010;@Jackson2006]. Previous analysis of *C. difficile* colonizing germ free mice under mono-associated conditions has indicated that *C. difficile* uses both sets of metabolites [@Janoir2013]; however, use of these metabolites in the context of a complex community of potential competitors has not been observed previously. This analysis indicated that these metabolites may be an integral component of the nutrient niche for *C. difficile*.

We next focused on the 25 highest scoring metabolites in each infection condition to screen for those metabolites that were important to *C. difficile* in each environment (Fig. 4B). The resulting groups of metabolites all contained at least one known carbohydrate growth substrate of *C. difficile* [@Wilson1988;@Ng2013;@Theriot2014]. These included sorbitol (streptomycin), mannitol (cefoperazone), and salicin (clindamycin), which are likely introduced by the animal's diet and N-acetylneuraminic acid (germ free), which is an amino sugar that is likely obtained from the animal's mucin layer [@Almagro-Moreno2009]. The concentration of N-acetylneuraminic acid was previously shown to be liberated by other bacterial species to the benefit of *C. difficile* [@Ng2013]. Considering that N-acetylneuraminic acid had one of the higher importance scores in our ex-germ free animals, this suggests that *C. difficile* does not require other bacteria to liberate it from the mucus layer. Furthermore, in ex-germ free mice where no other competitors are present, our model indicated that *C. difficile* was more likely to acquire numerous amino acids from the environment, presumably, instead of going through synthesizing them. These data supported the hypothesis that *C. difficile* exploited alternative nutrient sources between the susceptible environments it colonized.


**Important metabolites from *in vivo* network analysis are *C. difficile* growth substrates**
To further validate the results of our metabolic model, we tested the ability of *C. difficile* to grow *in vitro* with the carbon sources that had the highest importance scores (Fig. 4C). We supplemented a modified defined *C. difficile* minimal media with the carbohydrates that had high importance scores either across all conditions or within each condition [@Theriot2014]. Because *C. difficile* is auxotrophic for several amino acids, including proline [@Karasawa1995], it was necessary to include them although *C. difficile* could ferment them to support modest growth. This focused our analysis on carbohydrates and made media lacking carbohydrates but containing amino acids the most effective negative control. Supporting the results of our metabolic model and importance scores, we observed significant growth by *C. difficile* (all *P* < `r max(experimental_cp)`) on N-acetylglucosamine (shared; Max OD~600~ = `r acetylglucosamine_max_od`), mannitol (cefoperazone; Max OD~600~ = `r mannitol_max_od`), salicin (clindamycin; Max OD~600~ = `r salicin_max_od`), and N-acetylneuraminate (GF; Max OD~600~ = `r acetylneuraminate_max_od`). Galactitol was also tested as it was predicted to be important in streptomycin-treated animals; however, it had a nearly identical growth patterns to the no carbohydrate control (Table S4). These results supported the ability of our network algorithm to identify likely growth substrates and provided additional evidence that *C. difficile* was well-suited to adapt its metabolism toward specific nutrients. ~~These data also suggested a potential hierarchy in carbon source preference, but requires additional investigations to state conclusively.~~ [Comment: I'm sure you're right, but I don't see the evidence for it here and you certainly haven't articulated why you think there's a hierarchy]



```{r figure_5, echo=F, warning=F, message=F, results='hide', cache=F}


metabolome <- read.delim('data/wetlab_assays/metabolomics.tsv', sep='\t', header=TRUE, row.names=1)
metabolome <- metabolome[, !colnames(metabolome) %in% c('CefC5M2','StrepC4M1')] # Remove possible contamination
metadata <- read.delim('data/metadata.tsv', sep='\t', header=T, row.names=1)
metadata <- metadata[!rownames(metadata) %in% c('CefC5M2','StrepC4M1'), ] # Remove possible contamination
metadata$cage <- NULL
metadata$mouse <- NULL
metadata$gender <- NULL
metadata$type <- NULL
metabolome$SUPER_PATHWAY <- NULL
metabolome$SUB_PATHWAY <- NULL
metabolome$PUBCHEM <- NULL
metabolome$KEGG <- NULL
metabolome <- as.data.frame(t(metabolome))
metabolome <- merge(metadata, metabolome, by='row.names')
rownames(metabolome) <- metabolome$Row.names
metabolome$Row.names <- NULL
rm(metadata)
acetylglucosamine <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('N-acetylglucosamine/N-acetylgalactosamine')))]
acetylglucosamine_cef <- subset(acetylglucosamine, abx == 'cefoperazone')
acetylglucosamine_cef$abx <- NULL
colnames(acetylglucosamine_cef) <- c('infection', 'substrate')
acetylglucosamine_strep <- subset(acetylglucosamine, abx == 'streptomycin')
acetylglucosamine_strep$abx <- NULL
colnames(acetylglucosamine_strep) <- c('infection', 'substrate')
acetylglucosamine_clinda <- subset(acetylglucosamine, abx == 'clindamycin')
acetylglucosamine_clinda$abx <- NULL
colnames(acetylglucosamine_clinda) <- c('infection', 'substrate')
acetylglucosamine_gf <- subset(acetylglucosamine, abx == 'germfree')
acetylglucosamine_gf$abx <- NULL
colnames(acetylglucosamine_gf) <- c('infection', 'substrate')
rm(acetylglucosamine)
proline <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('proline')))]
proline_cef <- subset(proline, abx == 'cefoperazone')
proline_cef$abx <- NULL
colnames(proline_cef) <- c('infection', 'substrate')
proline_strep <- subset(proline, abx == 'streptomycin')
proline_strep$abx <- NULL
colnames(proline_strep) <- c('infection', 'substrate')
proline_clinda <- subset(proline, abx == 'clindamycin')
proline_clinda$abx <- NULL
colnames(proline_clinda) <- c('infection', 'substrate')
proline_gf <- subset(proline, abx == 'germfree')
proline_gf$abx <- NULL
colnames(proline_gf) <- c('infection', 'substrate')
rm(proline)
mannitolsorbitol <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('mannitol/sorbitol')))]
mannitolsorbitol_cef <- subset(mannitolsorbitol, abx == 'cefoperazone')
mannitolsorbitol_cef$abx <- NULL
colnames(mannitolsorbitol_cef) <- c('infection', 'substrate')
mannitolsorbitol_strep <- subset(mannitolsorbitol, abx == 'streptomycin')
mannitolsorbitol_strep$abx <- NULL
colnames(mannitolsorbitol_strep) <- c('infection', 'substrate')
rm(mannitolsorbitol)
salicin <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('salicylate')))]
salicin_clinda <- subset(salicin, abx == 'clindamycin')
salicin_clinda$abx <- NULL
colnames(salicin_clinda) <- c('infection', 'substrate')
rm(salicin)
acetylneuraminate <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('N-acetylneuraminate')))]
acetylneuraminate_gf <- subset(acetylneuraminate, abx == 'germfree')
acetylneuraminate_gf$abx <- NULL
colnames(acetylneuraminate_gf) <- c('infection', 'substrate')
rm(acetylneuraminate)
galactitol <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('galactitol_(dulcitol)')))]
galactitol_strep <- subset(galactitol, abx == 'streptomycin')
galactitol_strep$abx <- NULL
colnames(galactitol_strep) <- c('infection', 'substrate')
rm(galactitol)
rm(metabolome)

acetylglucosamine_p <- round(p.adjust(c(wilcox.test(subset(acetylglucosamine_strep, infection=='630')[,2], subset(acetylglucosamine_strep, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine_cef, infection=='630')[,2], subset(acetylglucosamine_cef, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine_clinda, infection=='630')[,2], subset(acetylglucosamine_clinda, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine_gf, infection=='630')[,2], subset(acetylglucosamine_gf, infection=='mock')[,2], exact=F)$p.value), method='BH'), digits=3)
acetylglucosamine_p_cef <- acetylglucosamine_p[1]
acetylglucosamine_p_clinda <- acetylglucosamine_p[2]
acetylglucosamine_p_gf <- acetylglucosamine_p[3]

proline_p <- round(p.adjust(c(wilcox.test(subset(proline_strep, infection=='630')[,2], subset(proline_strep, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(proline_cef, infection=='630')[,2], subset(proline_cef, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(proline_clinda, infection=='630')[,2], subset(proline_clinda, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(proline_gf, infection=='630')[,2], subset(proline_gf, infection=='mock')[,2], exact=F)$p.value), method='BH'), digits=3)
proline_p_cef <- proline_p[1]
proline_p_clinda <- proline_p[2]
proline_p_gf <- proline_p[3]

galactitol_p_strep <- round(wilcox.test(subset(galactitol_strep, infection=='630')[,2], subset(galactitol_strep, infection=='mock')[,2], exact=F)$p.value, digits=3)

mannitolsorbitol_p <- round(p.adjust(c(wilcox.test(subset(mannitolsorbitol_strep, infection=='630')[,2], subset(mannitolsorbitol_strep, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(mannitolsorbitol_cef, infection=='630')[,2], subset(mannitolsorbitol_cef, infection=='mock')[,2], exact=F)$p.value), method='BH'), digits=3)
mannitolsorbitol_p_strep <- mannitolsorbitol_p[1]
mannitolsorbitol_p_cef <- mannitolsorbitol_p[2]

salicin_p_clinda <- round(wilcox.test(subset(salicin_clinda, infection=='630')[,2], subset(salicin_clinda, infection=='mock')[,2], exact=F)$p.value, digits=3)

acetylneuraminate_p_gf <- round(wilcox.test(subset(acetylneuraminate_gf, infection=='630')[,2], subset(acetylneuraminate_gf, infection=='mock')[,2], exact=F)$p.value, digits=3)

```


```{r figure_S6, echo=F, warning=F, message=F, results='hide', cache=F}

# Read in data
metabolome <- read.delim('data/wetlab_assays/metabolomics.tsv', sep='\t', header=TRUE, row.names=1)
metabolome <- metabolome[, !colnames(metabolome) %in% c('CefC5M2','StrepC4M1')] # Remove possible contamination
metadata <- read.delim('data/metadata.tsv', sep='\t', header=T, row.names=1)
metadata <- metadata[!rownames(metadata) %in% c('CefC5M2','StrepC4M1'), ] # Remove possible contamination

# Merge
metadata$cage <- NULL
metadata$mouse <- NULL
metadata$gender <- NULL
metadata$type <- NULL
metabolome$SUPER_PATHWAY <- NULL
metabolome$SUB_PATHWAY <- NULL
metabolome$PUBCHEM <- NULL
metabolome$KEGG <- NULL
metabolome <- as.data.frame(t(metabolome))
metabolome <- merge(metadata, metabolome, by='row.names')
rownames(metabolome) <- metabolome$Row.names
metabolome$Row.names <- NULL
rm(metadata)

# Subset metabolomics
metabolome <- subset(metabolome, infection == 'mock')
metabolome$infection <- NULL
acetylglucosamine <- metabolome[, c(1,which(colnames(metabolome) %in% c('N-acetylglucosamine/N-acetylgalactosamine')))]
colnames(acetylglucosamine) <- c('abx', 'substrate')
acetylglucosamine$abx <- factor(acetylglucosamine$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
glycine <- metabolome[, c(1,which(colnames(metabolome) %in% c('glycine')))]
colnames(glycine) <- c('abx', 'substrate')
glycine$abx <- factor(glycine$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
proline <- metabolome[, c(1,which(colnames(metabolome) %in% c('proline')))]
colnames(proline) <- c('abx', 'substrate')
proline$abx <- factor(proline$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
mannitolsorbitol <- metabolome[, c(1,which(colnames(metabolome) %in% c('mannitol/sorbitol')))]
colnames(mannitolsorbitol) <- c('abx', 'substrate')
mannitolsorbitol$abx <- factor(mannitolsorbitol$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
salicin <- metabolome[, c(1,which(colnames(metabolome) %in% c('salicylate')))]
salicin$abx <- factor(salicin$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
acetylneuraminate <- metabolome[, c(1,which(colnames(metabolome) %in% c('N-acetylneuraminate')))]
colnames(acetylneuraminate) <- c('abx', 'substrate')
acetylneuraminate$abx <- factor(acetylneuraminate$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
galactitol <- metabolome[, c(1,which(colnames(metabolome) %in% c('galactitol_(dulcitol)')))]
colnames(galactitol) <- c('abx', 'substrate')
galactitol$abx <- factor(galactitol$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
hydroxyproline <- metabolome[, c(1,which(colnames(metabolome) %in% c('trans-4-hydroxyproline')))]
colnames(hydroxyproline) <- c('abx', 'substrate')
hydroxyproline$abx <- factor(hydroxyproline$abx, levels=c('none','streptomycin','cefoperazone','clindamycin','germfree'))
rm(metabolome)

acetylglucosamine_mock_p <- round(p.adjust(c(wilcox.test(subset(acetylglucosamine, abx=='none')[,2], subset(acetylglucosamine, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine, abx=='none')[,2], subset(acetylglucosamine, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine, abx=='none')[,2], subset(acetylglucosamine, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylglucosamine, abx=='none')[,2], subset(acetylglucosamine, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
acetylglucosamine_mock_p_strep <- acetylglucosamine_mock_p[1]
acetylglucosamine_mock_p_cef <- acetylglucosamine_mock_p[2]
acetylglucosamine_mock_p_clinda <- acetylglucosamine_mock_p[3]
acetylglucosamine_mock_p_gf <- acetylglucosamine_mock_p[4]

acetylneuraminate_mock_p <- round(p.adjust(c(wilcox.test(subset(acetylneuraminate, abx=='none')[,2], subset(acetylneuraminate, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylneuraminate, abx=='none')[,2], subset(acetylneuraminate, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylneuraminate, abx=='none')[,2], subset(acetylneuraminate, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(acetylneuraminate, abx=='none')[,2], subset(acetylneuraminate, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
acetylneuraminate_mock_p_strep  <- acetylneuraminate_mock_p[1]
acetylneuraminate_mock_p_cef  <- acetylneuraminate_mock_p[2]
acetylneuraminate_mock_p_clinda  <- acetylneuraminate_mock_p[3]
acetylneuraminate_mock_p_gf  <- acetylneuraminate_mock_p[4]

proline_mock_p <- round(p.adjust(c(wilcox.test(subset(proline, abx=='none')[,2], subset(proline, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(proline, abx=='none')[,2], subset(proline, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(proline, abx=='none')[,2], subset(proline, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(proline, abx=='none')[,2], subset(proline, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
proline_mock_p_strep <- proline_mock_p[1]
proline_mock_p_cef <- proline_mock_p[2]
proline_mock_p_clinda <- proline_mock_p[3]
proline_mock_p_gf <- proline_mock_p[4]

glycine_mock_p <- round(p.adjust(c(wilcox.test(subset(glycine, abx=='none')[,2], subset(glycine, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine, abx=='none')[,2], subset(glycine, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine, abx=='none')[,2], subset(glycine, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine, abx=='none')[,2], subset(glycine, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
glycine_mock_p_strep <- glycine_mock_p[1]
glycine_mock_p_cef <- glycine_mock_p[2]
glycine_mock_p_clinda <- glycine_mock_p[3]
glycine_mock_p_gf <- glycine_mock_p[4]

hydroxyproline_mock_p <- round(p.adjust(c(wilcox.test(subset(hydroxyproline, abx=='none')[,2], subset(hydroxyproline, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline, abx=='none')[,2], subset(hydroxyproline, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline, abx=='none')[,2], subset(hydroxyproline, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline, abx=='none')[,2], subset(hydroxyproline, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
hydroxyproline_mock_p_strep <- hydroxyproline_mock_p[1]
hydroxyproline_mock_p_cef <- hydroxyproline_mock_p[2]
hydroxyproline_mock_p_clinda <- hydroxyproline_mock_p[3]
hydroxyproline_mock_p_gf <- hydroxyproline_mock_p[4]

salicin_mock_p <- round(p.adjust(c(wilcox.test(subset(salicin, abx=='none')[,2], subset(salicin, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(salicin, abx=='none')[,2], subset(salicin, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(salicin, abx=='none')[,2], subset(salicin, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(salicin, abx=='none')[,2], subset(salicin, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
salicin_mock_p_strep <- salicin_mock_p[1]
salicin_mock_p_cef <- salicin_mock_p[2]
salicin_mock_p_clinda <- salicin_mock_p[3]
salicin_mock_p_gf <- salicin_mock_p[4]

mannitolsorbitol_mock_p <- round(p.adjust(c(wilcox.test(subset(mannitolsorbitol, abx=='none')[,2], subset(mannitolsorbitol, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(mannitolsorbitol, abx=='none')[,2], subset(mannitolsorbitol, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(mannitolsorbitol, abx=='none')[,2], subset(mannitolsorbitol, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(mannitolsorbitol, abx=='none')[,2], subset(mannitolsorbitol, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
mannitolsorbitol_mock_p_strep <- mannitolsorbitol_mock_p[1]
mannitolsorbitol_mock_p_cef <- mannitolsorbitol_mock_p[2]
mannitolsorbitol_mock_p_clinda <- mannitolsorbitol_mock_p[3]
mannitolsorbitol_mock_p_gf <- mannitolsorbitol_mock_p[4]

galactitol_mock_p <- round(p.adjust(c(wilcox.test(subset(galactitol, abx=='none')[,2], subset(galactitol, abx=='streptomycin')[,2], exact=F)$p.value,
           wilcox.test(subset(galactitol, abx=='none')[,2], subset(galactitol, abx=='cefoperazone')[,2], exact=F)$p.value,
           wilcox.test(subset(galactitol, abx=='none')[,2], subset(galactitol, abx=='clindamycin')[,2], exact=F)$p.value,
           wilcox.test(subset(galactitol, abx=='none')[,2], subset(galactitol, abx=='germfree')[,2], exact=F)$p.value), method='BH'), digits=3)
galactitol_mock_p_strep <- galactitol_mock_p[1]
galactitol_mock_p_cef <- galactitol_mock_p[2]
galactitol_mock_p_clinda <- galactitol_mock_p[3]
galactitol_mock_p_gf <- galactitol_mock_p[4]

```

```{r figure_S7, echo=F, warning=F, message=F, results='hide', cache=F}

# Read in data
scfa <- read.delim('data/wetlab_assays/cef_acetate_630.txt', sep='\t', header=T)
metabolome <- read.delim('data/wetlab_assays/metabolomics.tsv', sep='\t', header=T, row.names=1)
metabolome <- metabolome[, !colnames(metabolome) %in% c('CefC5M2','StrepC4M1')] # Remove possible contamination
metadata <- read.delim('data/metadata.tsv', sep='\t', header=T, row.names=1)
metadata <- metadata[!rownames(metadata) %in% c('CefC5M2','StrepC4M1'), ] # Remove possible contamination

# Format the data
scfa$group <- factor(scfa$group, levels=c('infected','mock'))
scfa$acetate <- as.numeric(as.character(scfa$acetate))

# Subset for stats
mock <- as.numeric(scfa[scfa$group == 'mock', 2])
infected <- as.numeric(scfa[scfa$group == 'infected', 2])

# Subset untargeted metabolomics
metadata$cage <- NULL
metadata$mouse <- NULL
metadata$gender <- NULL
metadata$type <- NULL
metabolome$SUPER_PATHWAY <- NULL
metabolome$SUB_PATHWAY <- NULL
metabolome$PUBCHEM <- NULL
metabolome$KEGG <- NULL
metabolome <- as.data.frame(t(metabolome))
metabolome <- merge(metadata, metabolome, by='row.names')
rownames(metabolome) <- metabolome$Row.names
metabolome$Row.names <- NULL
rm(metadata)
glycine <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('glycine')))]
glycine_cef <- subset(glycine, abx == 'cefoperazone')
glycine_cef$abx <- NULL
colnames(glycine_cef) <- c('infection', 'substrate')
glycine_strep <- subset(glycine, abx == 'streptomycin')
glycine_strep$abx <- NULL
colnames(glycine_strep) <- c('infection', 'substrate')
glycine_clinda <- subset(glycine, abx == 'clindamycin')
glycine_clinda$abx <- NULL
colnames(glycine_clinda) <- c('infection', 'substrate')
glycine_gf <- subset(glycine, abx == 'germfree')
glycine_gf$abx <- NULL
colnames(glycine_gf) <- c('infection', 'substrate')
rm(glycine)
hydroxyproline <- metabolome[, c(1,2,which(colnames(metabolome) %in% c('trans-4-hydroxyproline')))]
hydroxyproline_cef <- subset(hydroxyproline, abx == 'cefoperazone')
hydroxyproline_cef$abx <- NULL
colnames(hydroxyproline_cef) <- c('infection', 'substrate')
hydroxyproline_strep <- subset(hydroxyproline, abx == 'streptomycin')
hydroxyproline_strep$abx <- NULL
colnames(hydroxyproline_strep) <- c('infection', 'substrate')
hydroxyproline_clinda <- subset(hydroxyproline, abx == 'clindamycin')
hydroxyproline_clinda$abx <- NULL
colnames(hydroxyproline_clinda) <- c('infection', 'substrate')
hydroxyproline_gf <- subset(hydroxyproline, abx == 'germfree')
hydroxyproline_gf$abx <- NULL
colnames(hydroxyproline_gf) <- c('infection', 'substrate')
rm(hydroxyproline)
rm(metabolome)

glycine_p <- round(p.adjust(c(wilcox.test(subset(glycine_strep, infection=='630')[,2], subset(glycine_strep, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine_cef, infection=='630')[,2], subset(glycine_cef, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine_clinda, infection=='630')[,2], subset(glycine_clinda, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(glycine_gf, infection=='630')[,2], subset(glycine_gf, infection=='mock')[,2], exact=F)$p.value), method='BH'), digits=3)
glycine_p_cef <- glycine_p[1]
glycine_p_clinda <- glycine_p[2]
glycine_p_gf <- glycine_p[3]

hydroxyproline_p <- round(p.adjust(c(wilcox.test(subset(hydroxyproline_strep, infection=='630')[,2], subset(hydroxyproline_strep, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline_cef, infection=='630')[,2], subset(hydroxyproline_cef, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline_clinda, infection=='630')[,2], subset(hydroxyproline_clinda, infection=='mock')[,2], exact=F)$p.value,
           wilcox.test(subset(hydroxyproline_gf, infection=='630')[,2], subset(hydroxyproline_gf, infection=='mock')[,2], exact=F)$p.value), method='BH'), digits=3)
hydroxyproline_p_cef <- hydroxyproline_p[1]
hydroxyproline_p_clinda <- hydroxyproline_p[2]
hydroxyproline_p_gf <- hydroxyproline_p[3]

acetate_p_cef <- wilcox.test(mock, infected, exact=F)$p.value

```

** *In vivo* metabolomic analysis supports that *C. difficile* consumes metabolites indicated by metabolic modeling.**
To further validate the results of our metabolic model, we tested the effect of *C. difficile* on the metabolite pool in antibiotic-treated and germ free mice. We used non-targeted ultra-performance liquid chromatography and mass spectrometry (UPLC-MS) to measure the relative *in vivo* concentrations of metabolites in the conditions investigated, with special attention to those highlighted by large importance scores. We tested whether the susceptible communities had significantly different concentrations of each metabolite relative to untreated SPF mice and whether the presence of *C. difficile* affected the metabolite composition.

First, we measured whether the relative concentration of important metabolites had a higher concentration in the susceptible environment relative to untreated SPF mice (Fig. S6). We found that the relative concentration of N-acetylglucosamine was actually significantly lower in all susceptible conditions (Fig. S6A; all *P* <= `r max(acetylglucosamine_mock_p)`). A similar trend was also seen for salicylate (Fig. S6F; all *P* <= `r max(salicin_mock_p)`), a product of salacin cleavage (salicin was not included in the panel) [Comment: Can you recalibrate the y-axis?]. The Stickland fermentation substrates proline (all *P* < `r max(proline_mock_p)`) and trans-4-hydroxyproline (all *P* < `r max(hydroxyproline_mock_p)`) were significantly higher in all susceptible environments tested (Fig. S6C and S6E). Among the cefoperazone-treated SPF and GF mice, we also found that N-acetylneuraminate (Fig. S6B), mannitol/sorbitol (Fig. S6G), galactitol (Fig. S6H), and glycine (Fig. S6D) were significantly higher in cefoperazone-treated SPF and GF mice (all *P* < 0.05). These results supported the assertion that antibiotic treatment opened nutrient niches that *C. difficile* was able to exploit for its growth.

Second, we measured the difference in the concentrations of important metabolites decreased *C. difficile* infection compared to mock infected mice (Fig. 5). Both groups of host-derived aminoglycans, N-acetylglucosamine/N−acetylgalactosamine (Fig. 5A) and N-acetylneuraminate (Fig. 5F), were significantly lower when the presence of *C. difficile* in ex-GF mice (*P* < `r acetylglucosamine_p_gf` and `r acetylneuraminate_p_gf`). No significant relationship was seen in either galactitol (Fig. 5C) or mannitol/sorbitol (Fig. 5D) (*P* > `r galactitol_p_strep` and `r min(mannitolsorbitol_p)`). In agreement with the previous results, we found that proline (Fig. 5A) and trans-4-hydroxyproline (Fig. S7B) were significantly lower in every *C. difficile* colonized environment (all *P* < `r max(c(proline_p,hydroxyproline_p))`). Glycine was lower in each condition following infection (Fig. S7A), with significant change in cefoperazone-pretreatment (*P* = `r glycine_p_cef`). These results strongly supported the hypothesis that amino acids are a primary energy source of *C. difficile* during infection. Together, these results provided validation for our metabolite importance algorithm as a method for predicting metabolites that are most likely consumed by a bacterium in a given environment, not just growth substrates. [Comment: This paragraph is a bit of a mess. It doesn't seem to agree with much of what you found elsewhere. Can this be explained by the preference hierarchy? For instance, if proline is around it might prefer to consume proline to anything else; but that would also squash the idea of using different substrates in different environments. This paragraph needs some deeper thinking...]

```{r figure_6, echo=F, warning=F, message=F, results='hide', cache=F}

# Metabolomic data
metabolome <- read.delim('data/wetlab_assays/metabolomics.tsv', sep='\t', header=T)
annotation <- metabolome[,1:5]
metabolome$SUPER_PATHWAY <- NULL
metabolome$SUB_PATHWAY <- NULL
metabolome$PUBCHEM <- NULL
metabolome <- subset(metabolome, KEGG != 'NA')
metabolome <- metabolome[match(unique(metabolome$KEGG), metabolome$KEGG),]
rownames(metabolome) <- metabolome$KEGG
metabolome$KEGG <- NULL
metabolome$BIOCHEMICAL <- NULL
annotation <- subset(annotation, KEGG != 'NA')
annotation <- annotation[match(unique(annotation$KEGG), annotation$KEGG),]
rownames(annotation) <- annotation$KEGG
annotation$KEGG <- NULL
annotation$SUB_PATHWAY <- NULL
annotation$PUBCHEM <- NULL
annotation$BIOCHEMICAL <- NULL
annotation$SUPER_PATHWAY <- gsub('_',' ', annotation$SUPER_PATHWAY)

# Combine with metadata
metadata <- read.delim('data/metadata.tsv', sep='\t', header=T, row.names=1)
metabolome <- merge(t(metabolome), metadata, by='row.names')
rm(metadata)
rownames(metabolome) <- metabolome$Row.names
metabolome$Row.names <- NULL
metabolome <- subset(metabolome, abx != 'none')
metabolome_630 <- subset(metabolome, infection == '630')
metabolome_630$cage <- NULL
metabolome_630$mouse <- NULL
metabolome_630$gender <- NULL
metabolome_630$type <- NULL
metabolome_630$infection <- NULL
metabolome_630 <- aggregate(metabolome_630[, 1:399], list(metabolome_630$abx), median)
rownames(metabolome_630) <- metabolome_630$Group.1
metabolome_630$Group.1 <- NULL
metabolome_630 <- as.data.frame(t(metabolome_630))
colnames(metabolome_630) <- c('cef_630','clinda_630','gf_630','strep_630')
metabolome_mock <- subset(metabolome, infection == 'mock')
metabolome_mock$cage <- NULL
metabolome_mock$mouse <- NULL
metabolome_mock$gender <- NULL
metabolome_mock$type <- NULL
metabolome_mock$infection <- NULL
metabolome_mock <- aggregate(metabolome_mock[, 1:399], list(metabolome_mock$abx), median)
rownames(metabolome_mock) <- metabolome_mock$Group.1
metabolome_mock$Group.1 <- NULL
metabolome_mock <- as.data.frame(t(metabolome_mock))
colnames(metabolome_mock) <- c('cef_mock','clinda_mock','gf_mock','strep_mock')
metabolome <- metabolome_mock / metabolome_630
metabolome <- merge(metabolome, annotation, by='row.names')
rownames(metabolome) <- metabolome$Row.names
metabolome$Row.names <- NULL
colnames(metabolome) <- c('cefoperazone_conc', 'clindamycin_conc', 'germfree_conc', 'streptomycin_conc', 'pathway')
rm(annotation, metabolome_mock, metabolome_630)

# Merge importances to one table
cef_importances <- read.delim('data/metabolic_models/cefoperazone_630.bipartite.files/importances.tsv', sep='\t', header=T, row.names=1)
cef_importances$p_value <- NULL
clinda_importances <- read.delim('data/metabolic_models/clindamycin_630.bipartite.files/importances.tsv', sep='\t', header=T, row.names=1)
clinda_importances$p_value <- NULL
clinda_importances$Compound_name <- NULL
strep_importances <- read.delim('data/metabolic_models/streptomycin_630.bipartite.files/importances.tsv', sep='\t', header=T, row.names=1)
strep_importances$p_value <- NULL
strep_importances$Compound_name <- NULL
gf_importances <- read.delim('data/metabolic_models/germfree_630.bipartite.files/importances.tsv', sep='\t', header=T, row.names=1)
gf_importances$p_value <- NULL
gf_importances$Compound_name <- NULL
importances <- merge(cef_importances, clinda_importances, by='row.names')
rownames(importances) <- importances$Row.names
importances$Row.names <- NULL
importances <- merge(importances, strep_importances, by='row.names')
rownames(importances) <- importances$Row.names
importances$Row.names <- NULL
importances <- merge(importances, gf_importances, by='row.names')
rownames(importances) <- importances$Row.names
importances$Row.names <- NULL
colnames(importances) <- c('Compound_name', 'cefoperazone_score', 'clindamycin_score', 'streptomycin_score', 'germfree_score')
importances$Compound_name <- gsub('_',' ', importances$Compound_name)
rm(cef_importances, clinda_importances, strep_importances, gf_importances)

# Merge metabolome medians and importance values
combined <- merge(importances, metabolome, by='row.names')
rownames(combined) <- combined$Row.names
combined$Row.names <- NULL
rm(importances, metabolome)

# Separate treatment groups
cef <- as.data.frame(cbind(combined$cefoperazone_score, combined$cefoperazone_conc, combined$pathway))
rownames(cef) <- rownames(combined)
colnames(cef) <- c('score', 'conc', 'pathway')
cef$name <- combined$Compound_name
cef$score <- as.numeric(as.vector(cef$score))
cef$conc <- as.numeric(as.vector(cef$conc))
cef <- subset(cef, cef[,1] != 0) # remove metabolites with 0 importance
clinda <- as.data.frame(cbind(combined$clindamycin_score, combined$clindamycin_conc, combined$pathway))
rownames(clinda) <- rownames(combined)
colnames(clinda) <- c('score', 'conc', 'pathway')
clinda$name <- combined$Compound_name
clinda$score <- as.numeric(as.vector(clinda$score))
clinda$conc <- as.numeric(as.vector(clinda$conc))
clinda <- subset(clinda, clinda[,1] != 0) # remove metabolites with 0 importance
strep <- as.data.frame(cbind(combined$streptomycin_score, combined$streptomycin_conc, combined$pathway))
rownames(strep) <- rownames(combined)
colnames(strep) <- c('score', 'conc', 'pathway')
strep$name <- combined$Compound_name
strep$score <- as.numeric(as.vector(strep$score))
strep$conc <- as.numeric(as.vector(strep$conc))
strep <- subset(strep, strep[,1] != 0) # remove metabolites with 0 importance
germfree <- as.data.frame(cbind(combined$germfree_score, combined$germfree_conc, combined$pathway))
rownames(germfree) <- rownames(combined)
colnames(germfree) <- c('score', 'conc', 'pathway')
germfree$name <- combined$Compound_name
germfree$score <- as.numeric(as.vector(germfree$score))
germfree$conc <- as.numeric(as.vector(germfree$conc))
germfree <- subset(germfree, germfree[,1] != 0)
combined <- rbind(cef, clinda, strep, germfree)

# Subset N-acetylglucosamine
strep_acetylglucosamine <- subset(strep, rownames(strep) %in% c('C00140'))
cef_acetylglucosamine <- subset(cef, rownames(cef) %in% c('C00140'))
clinda_acetylglucosamine <- subset(clinda, rownames(clinda) %in% c('C00140'))
germfree_acetylglucosamine <- subset(germfree, rownames(germfree) %in% c('C00140'))
combined_acetylglucosamine <- subset(combined, rownames(combined) %in% c('C00140'))

# Fit to general linear models and identify outliers (L1 regression)
strep_fit <- glm(conc ~ score, data=strep)
strep$residuals <- residuals(strep_fit)
strep$residuals <- (strep$residuals / sd(strep$residuals))^2
strep_outliers <- strep[strep$residuals > 1.5, ]
cef_fit <- glm(conc ~ score, data=cef)
cef$residuals <- residuals(cef_fit)
cef$residuals <- (cef$residuals / sd(cef$residuals))^2
cef_outliers <- cef[cef$residuals > 1.5, ]
clinda_fit <- glm(conc ~ score, data=clinda)
clinda$residuals <- residuals(clinda_fit)
clinda$residuals <- (clinda$residuals / sd(clinda$residuals))^2
clinda_outliers <- clinda[clinda$residuals > 1.5, ]
germfree_fit <- glm(conc ~ score, data=germfree)
germfree$residuals <- residuals(germfree_fit)
germfree$residuals <- (germfree$residuals / sd(germfree$residuals))^2
germfree_outliers <- germfree[germfree$residuals > 1.5, ]
combined_fit <- glm(conc ~ score, data=combined)
combined$residuals <- residuals(combined_fit)
combined$residuals <- (combined$residuals / sd(combined$residuals))^2
combined_outliers <- combined[combined$residuals > 1.5, ]

# Calculate stats
corr_m <- round(c(strep_fit$coefficients[[2]],cef_fit$coefficients[[2]],clinda_fit$coefficients[[2]],
        germfree_fit$coefficients[[2]],combined_fit$coefficients[[2]]), digits=3)
corr_m_strep <- corr_m[1]
corr_m_cef <- corr_m[2]
corr_m_clinda <- corr_m[3]
corr_m_gf <- corr_m[4]
corr_m_combined <- corr_m[5]

corr_r <- round(c(cor.test(strep[,1], strep[,2], method='spearman', exact=FALSE)$estimate,
        cor.test(cef[,1], cef[,2], method='spearman', exact=FALSE)$estimate,
        cor.test(clinda[,1], clinda[,2], method='spearman', exact=FALSE)$estimate,
        cor.test(germfree[,1], germfree[,2], method='spearman', exact=FALSE)$estimate,
        cor.test(combined[,1], combined[,2], method='spearman', exact=FALSE)$estimate), digits=3)
corr_r_strep <- corr_r[1]
corr_r_cef <- corr_r[2]
corr_r_clinda <- corr_r[3]
corr_r_gf <- corr_r[4]
corr_r_combined <- corr_r[5]

corr_p <- round(c(cor.test(strep[,1], strep[,2], method='spearman', exact=FALSE)$p.value,
        cor.test(cef[,1], cef[,2], method='spearman', exact=FALSE)$p.value,
        cor.test(clinda[,1], clinda[,2], method='spearman', exact=FALSE)$p.value,
        cor.test(germfree[,1], germfree[,2], method='spearman', exact=FALSE)$p.value,
        cor.test(combined[,1], combined[,2], method='spearman', exact=FALSE)$p.value), digits=3)
corr_p_strep <- corr_p[1]
corr_p_cef <- corr_p[2]
corr_p_clinda <- corr_p[3]
corr_p_gf <- corr_p[4]
corr_p_combined <- corr_p[5]

```

\\\

[ Comment: I think this entire section could go away. It doesn't address your nutrient-niche hypothesis and if it does, it only adds confusion and length. I think it would be more useful in the full-community analysis. The paper isn't about the model, it's about testing the nutrient-niche hypothesis. I also find the writing in this section impenetrable ]

~~**Multi-omic integration supports predictive value of network platform for identifying previously unknown metabolic patterns.**
Moving beyond specific metabolite analysis, we determined how well metabolite importance scores related to the relative concentration of all metabolites in a given environment. To most effectively combine metabolite importance scores with the affiliated concentration data, we decided to calculate the median change in relative concentration for each metabolite measured between corresponding mock and *C. difficile*-infected conditions. This was accomplished by dividing the median log~10~-scaled intensity of each metabolite in mock-infected mice by its corresponding intensity in *C. difficile*-infected mice to give the delta median scaled intensity for each metabolite. In this way, larger values indicated those metabolites that decreased in concentration when *C. difficile* colonized. This also resulted in a single value we could combine with importance scores for each metabolite and perform a linear correlation analysis. Positive correlations indicated that the metabolites predicted to be consumed by *C. difficile* through transcript-informed metabolic networks have a concomitant decrease in the metabolome. This relationship would also imply that those metabolites with negative importance scores and increased concentration in infected animals are being produced by *C. difficile*.~~

[ Comment: If I'm following you, your delta median scaled intensity is median(log10(metabolite_mock))/median(log10(metabolite_cdiff))]. Why not log10(median(metabolite_mock)/median(metabolite_cdiff))? The log10 should be a transformation to reign in very spread data. Also, the way you're doing it, it is impossible to get negative values - I think you are squishing together all the decreases between 0 and 1

set.seed(1)
metabolite_mock <- runif(10)
metabolite_cdiff <- metabolite_mock*0.1
metabolite_mock
metabolite_cdiff
median(log10(metabolite_mock))/median(log10(metabolite_cdiff))
#0.1814105
log10(median(metabolite_mock)/median(metabolite_cdiff))
#1

The value of 1 makes sense since this is really a 10-fold difference, which is the true difference between the mock and cdiff conditions.
]

~~We applied this approach first to a pooled analysis of all groups to maximize our ability to validate the predictive capacity of the modeling algorithm (Fig. 6A). This revealed a significant positive correlation (*P* = `r corr_p_combined`) and supported our method for identifying metabolites that are most likely to be consumed by *C. difficile* based on transcriptional activity. This technique also allowed us to analyze outliers from the generalized linear model to assess if they are explained by other known patterns of *C. difficile* biology. Applying least-squared residuals analysis, we found that 16 [Comment: needs r code] metabolites were determined to be outliers and were all classified 1 of 3 separate KEGG annotation categories; amino acids, carbohydrates, and nucleotides (Table S5). We went on to identify the majority subset of 12 outlier metabolites that were important and significantly decreased in their respective condition were annotated as growth substrates or growth enhancers of *C. difficile* [@Karasawa1995]. This included L-proline, L-threonine, N-acetyl-D-glucosamine, D-fructose, and galactitol. These results were interpreted as not a failure of our importance algorithm, but as a possible underestimation of enzyme efficiency for the consumption these metabolites which further supports our approach as a way to determine real *in vivo* metabolic strategies for *C. difficile*. We also explored those metabolites that were more likely to be produced based on importance scores but decreased during infection. These were the nucleotides adenine, deoxyadenosine, and xanthine. *C. difficile* has been shown to possess the capacity for their synthesis [@Karlsson2008], but also may have the ability to catabolize these purines to participate downstream in Stickland reactions [@Hartwich2012]. The findings here may indicate that *C. difficile* utilizes both environmental and synthesized pools of the molecules at a faster rate than they are produced by the bacterium alone and may reinforce the centrality of amino acid fermentation to *C. difficile* *in vivo* metabolism.~~

~~We then directed the analysis toward each individual infection condition tested. In streptomycin-pretreatment (Fig. 6B) and cefoperazone-pretreatment (Fig. 6C) the correlations did not achieve significance (*P* = `r corr_p_strep` and `r corr_p_cef`), but remained positive. Analysis of outliers revealed that in streptomycin-pretreatment both the host-derived aminoglycan N-acetyl-D-glucosamine and L-proline were implicated as most likely to be utilized by *C. difficile* while L-proline was the only growth substrate outlier in this way in cefoperazone-pretreated mice. Both clindamycin-pretreatment (Fig. 6D) and ex-GF mice (Fig. 6E) instead demonstrated a significant correlation between importance scores and delta median scaled intensity (*P* = `r corr_p_clinda` and `r corr_p_gf`). Although clindamycin-pretreatment was associated with most outlier metabolites, both groups also shared a similar trend of carbohydrates and Stickland fermentation substrates being important and decreased in infected conditions. Proline specifically was found to be both highly important and decreased in concentration across all infections. In combination with the previously mentioned patterns, this further supports Stickland fermentation as a core metabolic strategy of *C. difficile* in the gut which agrees with previous *in vitro* findings [@Jackson2006]. Together, these data supported network-derived importance scores as well as our hypothesis that *C. difficile* adapted portions of its carbon source metabolism in distinct susceptible environments in addition to certain highly consistent strategies.~~


### Discussion ~ rewrite.

Outline:
Paragraph 1. Summarize results. How do the results support our hypothesis?

Paragraph 2. How do our results build upon/support/conflict with previous C.diff studies?

Paragraph 3. Novelty and importance of metabolic modeling that incorporates transcriptomics and was supported by metabolomics.

Paragraph 4. Limitation of our results. Pooling of mice to have sufficient high quality RNA for library construction. Meh because we have 16S, metabolomics, and they agree with each other. Could also point out low level of variation between replicates.

Paragraph 5. Conclude with teaser for full community metabolic model and metabolome

Our results support the hypothesis that *C. difficile* adjusts its metabolism to fit growth substrate availability in susceptible gut environments. The specific carbon sources identified as important to *C. difficile* supported substantial growth *in vitro* and decreased in concentration *in vivo* after the pathogen colonized the susceptible communities. Increased substrate availability was presumably the result of concomitant decreases in the population of one or more competitors for those resources. Some important substrates did not follow similar trends in the metabolome, but it does not necessarily imply that *C. difficile* cannot make use of them. It could simply mean that they are in large excess or their rate of consumption could be slow. Further research is required to explore these interactions. Our method is also able to identify consistent trends in metabolism across environments. The finding that Stickland fermentation substrates were consistently among the highest scoring shared metabolites and that they consistently decreased in concentration upon *C. difficile* colonization strongly indicated that these metabolites are central to the nutritional strategy of *C. difficile*. It stands to reason that catabolism of amino acids would be a highly conserved strategy for *C. difficile* since they provide not only carbon and energy, but are also a source for nitrogen which is a limited resource in the mammalian lower GI tract [@Fuller1998]. This same principle could extend to host-derived aminoglycans from the mucus layer. Concentrations of N-acetyl-D-glucosamine were consistently decreased when groups of bacteria with the capacity to cleave it from mucus were either diminished or absent entirely [@Marcobal2013]. N-acetylneuraminate was only found to be highly important in ex-GF mice when no other competitors are present. This could suggest that *C. difficile* is less competitive for these desirable nutrients, but can only really access them when competition is at a minimum.


[Clipped from Results]As *C. difficile* represented less than 3.3% of the bacterial load in the ceca of the antibiotic-treated mice (Fig. S2C), a high degree of sequencing depth per sample was required to yield sufficient quantities of reads contributed by *C. difficile*. This prohibited our ability to sequence each experimental replicate individually and made it necessary to pool samples within each treatment group, and allowed for some degree of conservation of replicates for downstream analysis. Since these circumstances were unavoidable, we treated this as normalizing for the average expression of *C. difficile* in each environment. Microarray-based gene expression measurement was not a viable alternative as the amount of background orthologous transcription from other bacterial species would contribute greatly to non-specific binding and eliminate true *C. difficile* signal. Upon sequencing, approximately 300 million total raw paired-end reads were collected per pooled sample with an average of approximately 160 thousand reads attributed to *C. difficile* following all curation steps (refer to Methods). This corresponded to an average of approximately 40x coverage across the *C. difficile* str. 630 reference genome. While this value may be low compared to some transcriptional studies of *in vitro* bacterial monoculture or *in vivo* ex-GF monoassociation [@Antunes2012;@Janoir2013], it is not feasible to reach higher coverage in the context of a diverse bacterial community. These levels have been shown to be largely acceptable for similar analyses due to the focus on large-scale differences in specific annotated genes [@Haas2012]. Microarray-based gene expression analysis was not a viable alternative as the amount of background transcription from other bacterial species would cause background non-specific binding of orthologous transcript and eliminate any true *C. difficile* signal.


[Clipped from Results]To accomplish this we mapped normalized transcript abundances to the enzyme nodes in the network. As transcription and translation are coupled in bacteria, we hypothesized that we could incorporate whole transcriptome sequencing results into the metabolic model to impute active metabolism. Several models of bacterial protein expression determinants also suggest that intracellular concentration of mRNA is the strongest predictor for abundance of the corresponding protein products [@Guimaraes2014]. We were therefore confident in utilizing normalized transcript abundance as a proxy for enzyme levels.

[Clipped from Results]Using these results, we first assessed variability in the datasets generated for this study with replicates. The goal was to infer possible consistency within transriptomic sequencing efforts without similar replication. Extremely low and consistent sample variance was found in across all 16S and metabolomic replicates (Fig. S5). Since species abundance partially determines enzyme abundance, which in turn partially controls changes in the metabolome, these data reassured us that our conclusions for transcriptional differences were reliable.

Based on the evidence presented, resistance to colonization by *C. difficile* could be due to an inability of the pathogen to outcompete a collection of metabolic specialists in an intact community and separate classes of antibiotics differentially eliminate subsets of these groups. This concept may potentially explain the success of fecal microbial transplant in eliminating *C. difficile* infection. The wholesale installation of a diverse range of specialized metabolic strategies might be enough to outcompete *C. difficile* from previously sensitive gut environments. This may be particularly true with competitors for fermentable amino acids as our data also suggests that these are critical to *C. difficile* overall *in vivo* metabolic strategy. A previous study generated a mutant *C. difficile* in proline reductase, which allows proline to enter Stickland fermentation, to evaluate how necessary proline is for growth [@Bouillaut2013]. *C. difficile* was able to grow without defect due to hydroxyproline, a non-proteinogenic amino acid, replacing proline in Stickland fermentation by circumventing the proline reductase dependent step of the pathway. As hydroxyproline shares the exact same relationship in concentrations as proline (Fig. S7B), it stands to reason that *C. difficile* is utilizing both during the infection conditions tested. As glycine (Fig. S7A) can also similarly be utilized, the redundancy in this pathway strongly supports how imperative this type of metabolism is to *C. difficile* during infection.

Our modeling approach also may allow for the identification of emergent properties for the metabolism of *C. difficile* during infection. One example could be the appearance of CO~2~, an apparent metabolic end product, in the list of shared important metabolites. While this may be a shortcoming of the annotation, one group has posited that *C. difficile* may actually be autotrophic under certain conditions and could explain our results [@Kopke2013]. This may highlight that our method not only identify growth substrates, but also reports metabolites that are being removed from the environment. Although our modeling results are consistent with previously published work on the metabolism of *C. difficile*, there are potential limitations of this approach. Ultimately, the metabolite importance calculation is dependent on correct and existing gene annotation. In this regard it has been shown that the pathway annotations in KEGG are robust to missing elements [@Green2006], however this does not completely eliminate the possibility for this type of error. Due to the topology of the metabolic network, we were also unable to integrate stoichiometry for each reaction which may effect rates of consumption or production. In spite of these assumptions, the method outlined here underscores known elements of *C. difficile* biology and our subsequent *in vivo* metabolomic analysis supported predictions of nutrient niche plasticity. Only through integrative multi-omic analysis of *C. difficile* infection employing genomics, transcriptomics, and metabolomics were we able to uncover a much clearer image of *C. difficile*'s metabolism during infection in the context of a complex community of bacteria. In conclusion, *C. difficile* was able to optimize its nutritional strategy for separate carbohydrates in each colonized gut environment. We also found that Stickland fermentation substrates and host-derived glycans were conserved elements of *C. difficile*'s nutrient niche across distinct gut communities. Our results implicate that further considerations are needed for the design or targeted prebiotic and probiotic therapies to prevent or eliminate of *C. difficile* from a colonized gut.


###Materials and Methods

**Animal care and antibiotic administration.** Six-to-eight week-old GF C57BL/6 mice were obtained from a single breeding colony maintained at the University of Michigan and fed Laboratory Rodent Diet 5001 from LabDiet for all experiments. All animal protocols were approved by the University Committee on Use and Care of Animals at the University of Michigan and carried out in accordance with the approved guidelines. Specified SPF animals were administered either cefoperazone, streptomycin, or clindamycin [@Theriot2011; @Schubert2015] (Table 1). Cefoperazone (0.5 mg/mL) and streptomycin (5.0 mg/mL) were administered in distilled drinking water *ad libitum* for 5 days with 2 days recovery with untreated distilled drinking water prior to infection. Clindamycin (10 mg/kg) was given via intraperitoneal injection 24 hours before time of infection.

** *C. difficile* infection and necropsy.** All *C. difficile* strain 630 spores were prepared from a single large batch whose concentration was determined a week prior to challenge. On the day of challenge, 1×10^3^ *C. difficile* spores were administered to mice via oral gavage in phosphate-buffered saline (PBS) vehicle. Subsequent quantitative plating for cfu was performed to ensure correct dosage. Mock-infected animals were given an oral gavage of 100 μl PBS at the same time as those mice administered *C. difficile* spores. 18 hours following infection, mice were euthanized by carbon dioxide asphyxiation and necropsied to obtain the cecal contents. Two 100 μl aliquots were immediately flash frozen for later DNA extraction and toxin titer analysis respectively. A third 100 μl aliquot was quickly transferred to an anaerobic chamber for quantification of *C. difficile* abundance. The remaining content in the ceca (approximately 1 mL) was mixed with 1 mL of sterile PBS in a stainless steel mortar housed in a dry ice and ethanol bath. The cecal content of 9 mice from 3 cages was pooled into the mortar. Pooling cecal contents was necessary so that there would be a sufficient quantity of high quality rRNA-free RNA for deep sequencing. The pooled content was then finely ground and stored at -80° C for subsequent RNA extraction.

** *C. difficile* cultivation and quantification.** Cecal conent samples were weighed and serially diluted under anaerobic conditions (6% H, 20% CO~2~, 74% N~2~) with anaerobic PBS. Differential plating was performed to quantify both *C. difficile* spores and vegetative cells by plating diluted samples on CCFAE plates (fructose agar plus cycloserine, cefoxitin, and erythromycin) at 37° C for 24 hours under anaerobic conditions [@Wilson1982]. The germination agent taurocholate was omitted from these plates to quantify only vegetative cells. In parallel, undiluted samples were heated at 60° C for 30 minutes to eliminate vegetative cells and leave only spores [@Sorg2010]. These samples were serially diluted under anaerobic conditions in anaerobic PBS and plated on CCFAE with taurocholate at 37° C for 24 hours. Plating was simultaneously done for heated samples on CCFAE to ensure all vegetative cells had been eliminated.

** *C. difficile* toxin titer assay.** To quantify the titer of toxin in the cecum, a Vero cell rounding assay was performed [@Leslie2015]. Briefly, filtered-sterilized cecal content was serially diluted 1:5 in PBS [Comment: is that 1:4 or 1:5? (see next comment)]. As a control for toxin-mediated cell rounding the cecal content was diluted a further 1:2 by the addition of an equal volume of goat anti-toxin serum [Comment: this would be 1:1 ](T5000; TechLab). ~~Vero cells were grown to a confluent monolayer in DMEM (Dulbecco's Modified Eagle's medium), supplemented with 10% heat-inactivated fetal bovine serum and 1% penicillin-streptomycin. The cells were then pelleted by centrifugation at 1,000 rpm for 5 minutes. The media was removed and the cells were re-suspended in fresh media to a final concentration of 1×10^5^ cells per 90μL. 90μL of the cell suspension were seeded in each well of a 96-well plate and incubated at 37° C in a 5% CO~2~ humidified incubator for 4 hours. Following the incubation, cecal samples were added to the Vero cells and the plate was incubated overnight at 37° C.~~ [Comment: this seems like a lot of unnecessary methods description when you are going to give the protocol below. The terseness of what you have below for the 16S rRNA gene sequencing section is more appropriate ] Plates were viewed after 24 hours at 10x magnification for cell rounding. The cytotoxic titer was defined as the log~10~ transformed reciprocal of the highest dilution that produced rounding in 80% of the cells. A more detailed protocol with product information can be found at: https://github.com/jlleslie/Intraspecific_Competition/blob/master/methods/Verocell_ToxinActivity_Assay.Rmd [ Comment: Can you move a copy of this to your paper's repository within SchlossLab so that I don't have to worry about link rot? ]

**16S rRNA gene sequencing.** DNA was extracted from approximately 50 mg of cecal content from each mouse using the PowerSoil-htp 96 Well Soil DNA isolation kit (MO BIO Laboratories) and an epMotion 5075 automated pipetting system (Eppendorf). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers and sequenced as described previously using an Illumina MiSeq sequencer [@Kozich2013]. All 63 samples were sequenced on a single sequencing run.

**Sequence curation** The 16S rRNA gene sequences were curated using the mothur software package (v1.36), as described previously [@Kozich2013]. In short, paired-end reads were merged into contigs, screened for quality, aligned to SILVA 16S rRNA sequence database, and screened for chimeras. Sequences were classified using a naive Bayesian classifier trained against a 16S rRNA gene training set provided by the Ribosomal Database Project (RDP) [@Wang2007]. Curated sequences were clustered into operational taxonomic units (OTUs) using a 97% similarity cutoff with the average neighbor clustering algorithm. Each sample was rarefied to 2,500 sequences per sample to minimize the effects of uneven sampling.

**RNA extraction, library preparation, and sequencing** ~~To generate enough mRNA biomass contributed by *C. difficile*, we pooled cecal content from all mouse replicates into a single large isolation for each treatment group. Pooling was performed in a sterile stainless steel mortar resting in dry ice and a small amount of 100% ethanol. After all content for the given group was added, the sample was ground with a sterile pestle to a fine powder and scraped into a sterile 50 mL polypropylene conical tube. Samples were stored at -80° C until the time of extraction.~~ [Comment: you already said all of this above] Immediately before RNA extraction, 3 mL of lysis buffer (2% SDS, 16 mM EDTA and 200 mM NaCl) was heated for 5 minutes in a boiling water bath [@Lopez-Medina2011]. The hot lysis buffer was mixed with the frozen and ground cecal content by periodic vortexing for 5 minutes in the boiling water bath. An equal volume of 37° C acid phenol/chloroform was added to the cecal content lysate and incubated at 37° C for 10 minutes with periodic vortexing. The mixture was the centrifuged at 2,500 x g at 4° C for 15 minutes. The aqueous phase was then transferred to a sterile tube and an equal volume of acid phenol/chloroform was added. This mixture was vortexed and centrifuged at 2,500 x g at 4° for 5 minutes. The process [Comment: what "process"?] was repeated until aqueous phase was clear. The last extraction was performed with chloroform/isoamyl alcohol to remove the residual acid phenol followed by a isopropanol precipitation. The nucleic acids were resuspended in RNase-free water and treated with 2 μl of Turbo DNase for 30 minutes at 37° C. Purified RNA was retrieved using the Zymo Quick-RNA MiniPrep (Insert name of manufacturer). Removal of DNA was assessed by attempting to PCR amplify the V4 region of the 16S rRNA gene using 30 cycles [@Kozich2013]. The quality and integrity of the RNA was measured using the Agilent RNA 6000 Nano kit for total prokaryotic RNA. The Ribo-Zero Gold rRNA Removal Kit Epidemiology was then used to deplete prokaryotic and eukaryotic rRNA from the samples (Insert name of manufacturer). Prior to library construction, the quality and integrity of the RNA was assessed using the Agilent RNA 6000 Pico Kit. Stranded RNA-Seq libraries were made constructed with the TruSeq Total RNA Library Preparation Kit v2 (Insert name of manufacturer). The Agilent DNA High Sensitivity Kit was used to measure concentration and fragment size distribution before sequencing (Insert name of manufacturer). High-throughput sequencing was performed by the University of Michigan Sequencing Core in Ann Arbor, MI. Parallel copies of the libraries were sequenced across 4 lanes of an Illumina HiSeq 2500 using the 2x50 bp chemistry.

**Sequence curation, read mapping, and normalization.** Raw transcript sequencing read curation was performed in a two step process. Residual 5’ and 3’ Illumina adapter sequences were trimmed using CutAdapt [@Martin2013] on a per library basis. Reads were quality trimmed using Sickle [Insert bibtex reference [@Joshi2011]] on the default settings. An average [Comment: it's not clear what these were averaged over - then you say "total" is this the average for each library or the total over all libraries?] of approximately 300,000,000 total reads (both paired and orphaned) remained after quality trimming. Mapping of individual reads to the KEGG annotated nucleotide gene sequences of *Clostridium difficile* str. 630 was performed using Bowtie2's stringent settings [@Langmead2009; @Ogata1999]; approximately 1,600,000 reads from each sample mapped to the genome seqeunce. Optical and PCR duplicates were then removed using Picard MarkDuplicates (http://broadinstitute.github.io/picard/); approximately 150,000 reads per sample remained for our final analysis. The remaining mappings were converted to idxstats format using Samtools [@Li2009] and the read counts per gene were tabulated. Discordant pair mappings were discarded and counts were then normalized to read length and gene length to give a per base report of gene coverage. Unless indicated otherwise, each collection of reads was then 1000-fold iteratively subsampled to 90% of the lowest sequence total within each analysis, and a median expression value for each gene was calculated.[Comment: I don't understand this sentence. Why weren't they rarefied to 100% of the smallest library?]

**Reaction Annotation and Bipartite Network Construction.** The metabolism of *C. difficile* str. 630 was represented as a directed bipartite graph with both enzymes and metabolites as nodes. Briefly, models were semi-automatically constructed using KEGG (2016 edition) ortholog (KO) gene annotations to which transcripts had been mapped. Reactions that each KEGG ortholog mediate were extracted from ko_reaction.list located in /kegg/genes/ko/. KOs that do not mediate simple biochemical reactions (ex. mediate interactions of macromolecules) were omitted. Metabolites linked to each reaction were retrieved from reaction_mapformula.lst file located in /kegg/ligand/reaction/ from the KEGG release. Those reactions that did not have annotations for the chemical compounds the interact with were discarded. Metabolites were then associated with each enzyme and the directionality and reversibility of each biochemical conversion was also saved. This process was repeated for all enzymes in the given bacterial genome, with each enzyme and metabolite node only appearing once. The resulting data structure was an associative array of enzymes associated with lists of both categories of substrates (input and output), which could then be represented as a bipartite network. The final metabolic network of *C. difficile* str. 630 contained a total of 1205 individual nodes (447 enzymes and 758 substrates) with 2135 directed edges. Transcriptomic data was then re-associated with the respective enzyme nodes prior to substrate importance calculations. Betweenness-centrality and overall closeness centralization indices were calculated using the igraph R package found at http://igraph.org/r/.

**Metabolite Importance Calculation.** To calculate the importance of a given metabolite (m), we used rarefied transcript abundances mapped to respective enzyme nodes. This was represented by t~o~ and t~i~ to designate if an enzyme created or utilized m. The first step was to calculate the average expression of enzymes for reactions that either created a given metabolite (i) or consumed that metabolite (ii). For each direction, the sum of transcripts for enzymes connecting to a metabolite were divided by the number of contributing edges (e~o~ or e~i~) to normalize for highly connected metabolite nodes. Next the raw metabolite importance score was calculated by subtracting the creation value from the consumption value to weight for metabolites that are likely acquired exogenously. The difference was log~2~ transformed for comparability between scores of individual metabolites. This resulted in a final value that reflected the likelihood a metabolite was acquired from the environment. Untransformed scores that already equaled to 0 were ignored and negative values were accounted for by transformation of the absolute value then multiplied by -1. These methods have been written into a single python workflow, along with supporting reference files, and is presented as bigSMALL (BacterIal Genome-Scale Metabolic models for AppLied reverse ecoLogy) available in a public Github repository at https://github.com/mjenior/bigsmall. [Comment: move into the SchlossLab project account]

**Transcriptome Randomization and Probability Distribution Comparison.** We developed a Monte Carlo-based analysis to identify those metabolite importance scores that were significantly higher than we would expect if the transcript abundances were randomly assigned to enzyme nodes. Transcript abundances were randomly assigned to enzyme nodes 10,000 times and for each bootstrapped replicate, we re-calculated the node's importance score. Those nodes with an observed importance score that was higher than 95% of the bootstrapped replicates were considered significant.

~~**Measuring *in vivo* concentrations of acetate.** Cecal contents were flash frozen in liquid nitrogen at the time of necropsy and subjected to short chain fatty acid quantification analysis using GC-MS (gas chromatography–mass spectrometry) [@Theriot2014]. All assays were performed at the Michigan Regional Comprehensive Metabolomics Resource Core in Ann Arbor, MI.~~ [Comment: Unless I'm missing something, you don't talk about these data]

**Anaerobic *in vitro* *C. difficile* growth curves.** The carbon-free variation of *C. difficile* Basal Defined Medium (NCMM) was prepared as previously described [@Theriot2014]. Individual carbohydrate sources were added at a final concentration of 5 mg/mL and pair-wise carbohydrate combinations were added at 2.5 mg/mL each (5 mg/mL total). A solution of the required amino acids was made separately and added when noted at identical concentrations to the same study. 245 μl of final media mixes were added to a 96-well sterile clear-bottom plate. A rich media growth control was also included, consisting of liquid Brain-Heart Infusion + 0.5% cysteine. All culturing and growth measurement were performed anaerobically in a Coy Type B Vinyl Anaerobic Chamber (3.0% H, 5.0% CO~2~, 92.0% N, 0.0% O~2~). *C. difficile* str. 630 was grown for 14 hours at 37° C in 3 mL BHI + 0.5% cysteine. Cultures were then centrifuged at 2000 rpm for 5 minutes and resulting pellets were then washed twice with sterile, anaerobic 1 × phosphate-buffered saline (PBS). Washed pellets were resuspended in 3 mL more PBS and 5 μl of prepped culture was added the each growth well of the plate containing media. The plate was then placed in a Tecan Sunrise plate reader, heated to 37° C. Plates were incubated for 24 hours with automatic optical density readings at 600 nm taken every 30 minutes. OD~600~ values were normalized to readings from wells containing sterile media of the same type at equal time of incubation. Growth rates and other curve metrics were determined by differentiation analysis of the measured OD~600~ over time in R to obtain the slope at each time point.

**Quantification of *in vivo* metabolite relative concentrations.** Metabolomic analysis of the flash-frozen cecal contents was performed by Metabolon (Durham, NC). The undirected metabolomic analysis utilized a Waters ACQUITY ultra-performance liquid chromatography (UPLC) and a Thermo Scientific Q-Exactive high resolution/accurate mass spectrometer interfaced with a heated electrospray ionization (HESI-II) source and Orbitrap mass analyzer at 35,000 mass resolution. Samples were dried then reconstituted in solvents compatible to each of the four methods [Comment: I count two methods. In the Results section you describe the UPLC results - what about the MS data?]. The first, in acidic positive conditions using a C18 column (Waters UPLC BEH C18-2.1x100 mm, 1.7 µm) using water and methanol, containing 0.05% perfluoropentanoic acid (PFPA) and 0.1% formic acid (FA). The second method was identical to the first but was chromatographically optimized for more hydrophobic compounds. The third approach utilized a basic negative ion optimized conditions using a separate dedicated C18 column. Basic extracts were gradient eluted from the column using methanol and water, however with 6.5mM Ammonium Bicarbonate at pH 8. Samples were then analyzed via negative ionization following elution from a hydrophilic interaction chromatography column (Waters UPLC BEH Amide 2.1x150 mm, 1.7 µm) using a gradient consisting of water and acetonitrile with 10mM Ammonium Formate, pH 10.8. The MS analysis alternated between MS and data-dependent MS n scans using dynamic exclusion. The scan range varied slighted between methods but covered 70-1000 m/z. Library matches for each compound were checked for each sample and corrected if necessary. Peaks were quantified using area under the curve.

**Statistical methods.** All statistical analyses were performed using R (v.3.2.0). Significant differences between community structure of treatment groups from 16S rRNA gene sequencing were determined with AMOVA in the mothur software package. Significant differences of inverse Simpson diversity, CFU counts, toxin titer, and metabolite concentrations were determined by Wilcoxon signed-rank test with Benjamini-Hochberg correction. Undetectable points used the limit of detection for all statistical calculations [Comment: This should be something like 1/2 the LOD - the value doesn't matter since you're using a non-parametric test]. Significant differences for growth curves compared to no carbohydrate control (+ amino acids) were calculated using 2-way ANOVA [why 2-way? isn't it a single classification ANOVA?] with Benjamini-Hochberg correction. ~~Outlier metabolites in linear correlation analysis were identified with a 1.5 minimum cutoff for squared Studentized residual analysis of each individual generalized linear model.~~


###Funding Information

This work was supported by funding from the National Institutes of Health to PDS (R01GM099514, P30DK034933, U19AI09087, and U01AI124255), VBY (P30DK034933, U19AI09087, and U01AI124255), a Translational Research Education Certificate grant to JLL (MICHR; UL1TR000433), and was partially supported by a predoctoral fellowship from the Cellular Biotechnology Training Program to MLJ (T32GM008353).


###Acknowledgements

The authors thank Charles Koumpouras for assistance with DNA extractions and metabolomic sample preparation. We would also like to acknowledge members of the University of Michigan Germfree Mouse Center, University of Michigan Sequencing Core, and Metabolon for their assistance in experimental design, execution, and data collection. Pooled and quality trimmed transcriptomic read data and experiment metadata are available through the NCBI Sequence Read Archive (SRA; PRJNA354635). Data processing steps for beginning from raw sequence data to the final manuscript are hosted at http://www.github.com/SchlossLab/Jenior_Modeling_mBio_2016.

**Author Affiliations**
**Department of Microbiology and Immunology, University of Michigan, Ann Arbor, Michigan.**
Matthew L. Jenior, Jhansi L. Leslie, and Patrick D. Schloss Ph.D.

**Department of Internal Medicine and Infectious Diseases, University of Michigan Medical Center, Ann Arbor, Michigan.**
Vincent B. Young M.D. Ph.D.

**Author Contributions**
M.L.J. conceived, designed and performed experiments, analyzed data, and drafted the manuscript. J.L.L. performed experiments and analyzed data. V.B.Y. contributed to the manuscript. P.D.S. interpreted data and contributed to the manuscript.
The authors declare no conflicts of interest.

**Corresponding author**
Correspondence to [Patrick D. Schloss](pdschloss@umich.edu)


###Figure Legends

**Table 1 | Antibiotics used during *C. difficile* infection models.**

**Figure 1 | *C. difficile* sporulation and toxin activity quantification after 18 hours of infection.** **(A)** Vegetative *C. difficile* density. **(B)** *C. difficile* spore density. **(C)** Toxin titer from cecal content measured by activity in Vero cell rounding assay. Ex-GF mice displayed significantly more spore cfu and toxin activity than all other groups (all *P* < `r gf_both_p`). Untreated mice had no detectable *C. difficile* or toxin activity (all *P* < 0.01). Dotted lines denote limits of detection (LOD), with undetectable points shown just below for clarity. Nine mice were included in each treatment group.

**Figure 2 | Ratios of *in vivo* expression for specific *C. difficile* metabolic pathways across antibiotic-pretreated mice.** Each point in the ternary plot represents a unique gene from the annotated genome of *C. difficile*. Position reflects the ratio of median rarefied transcript abundance for that gene between the three colonized antibiotic pretreatment models. Genes from specific metabolic pathways of interest are colored and all other genes not included in these groups are colored gray. **(A)** Size of highlighted points represents the largest transcript abundance among the antibiotic pretreatments for each individual gene. Metabolic categories are also plotted separately in **(B-I)**. Genes, annotations, and normalized transcript abundances can be found in Table S1. Refer to Fig. S4 for additional explanation of figure interpretation.

**Figure 3 | Genome-scale bipartite metabolic modeling results using the transcriptome of *C. difficile* in each colonized environment.** **(A)** Enzyme node sizes reflect the levels of detectable transcript from each gene. Importance algorithm components: (I) average transcription of reactions consuming a metabolite, (II) average transcription of reactions producing a metabolite, and (III) difference of consumption and production. **(B)** The expanded window displays a partial example of D-fructose importance calculation. Values in the red nodes represent normalized transcript reads mapping to enzymes. **(C)** Example 10,000-fold bootstrapping results corresponding to a significant importance score for **m**. [Comment: You draw C as if it were a two-tailed test, but didn't you do a one tailed test? For instance, the description of the next figure says <5%, not <2.5%]

**Figure 4 | Results from network-based metabolite importance calculation and *in vitro* *C. difficile* growth with important carbon sources.** **(A)** Median shared metabolites among the 50 highest scoring and significant metabolites from each condition. **(B)** Uniquely important metabolites from each treatment group. **(C)** 18 hour *C. difficile* str. 630 *in vitro* growth curves validating identified growth nutrients from network analysis. All statistical comparison was performed in comparison to no carbohydrate control (*P* < 0.001).

**Figure 5 | Change in *in vivo* concentrations following *C. difficile* infection of select metabolites indicated as important by metabolic modeling.** Paired metabolites were quantified simultaneously as the only differ by chirality making differentiation impossible. Significant differences (*P* < 0.05) are indicated with an asterisk.

**Figure 6 | Correlation of change in metabolite concentration following *C. difficile* colonization with importance scores derived from transcript-informed metabolic modeling.** Spearman correlations [Comment: if you're applying a GLM, why not Pearson?] for general linear models of change in metabolome due to *C. difficile* infection (n = 9 per group) and importance scores from metabolic modeling for each metabolite. Outlier metabolites are labeled for each correlation. Outlier metabolites are those outside the gray swath and were identified using Studentized-least squared residual analysis using a cutoff of more than 1.5.

**Supplementary Figure 1 | Experimental time lines for mouse model pretreatments and *C. difficile* infection.** 9 wild-type C57BL/6 mice across 3 cages were included in each treatment group. **(A)** Streptomycin or **(B)** cefoperazone administered *ad libitum* in drinking water for 5 days with 2 days recovery with untreated drinking water before infection, **(C)** a single clindamycin intraperitoneal injection one day prior to infection, or **(D)** no antibiotic pretreatment (for both SPF control and GF mice). If no antibiotics were administered in the drinking water, mice were given untreated drinking water for the duration of the experiment beginning 7 days prior to infection. At the time of infection, mice were challenged with 1×10^3^ *C. difficile* str. 630 spores. Euthanization and necropsy was done 18 hours post-challenge and cecal content was then collected. [Comment: should add a red triangle to the legend and change the heading for C to "Clindamycin (SPF)"]

**Supplementary Figure 2 | Analysis of bacterial community structure resulting from antibiotic treatment.** Results from 16S rRNA gene amplicon sequencing from bacterial communities of cecal content in both mock-infected and *C. difficile* 630-infected animals 18 hours post-infection across pretreatment models. **(A)** Non-metric multidimensional scaling (NMDS) ordination based on Theta~YC~ distances for the gut microbiome of all SPF mice used in these experiments (n = 63). All treatment groups were significantly different from each other groups by AMOVA (*P* < 0.001). **(B)** Inverse Simpson diversity for each cecal community from the mice in (A). Cecal communities from mice not treated with any antibiotics were significantly more diverse than any antibiotic-treated condition (*P* < 0.001). ~~**(C)** Representation of reads contributed by *C. difficile* in each sequenced condition.~~ [Comment: I think your y-axis label is wrong - there shouldn't have been much/any rRNA in your libraries - do you mean total reads? Not clear how you got a range of 630 reads in the infected columns. You also don't need a statistical test to say that 0.074% is significantly different from 50%, especially since you only have 1 sample. I would ditch panel C].

**Supplementary Figure 3 | Select *C. difficile* gene set expression compared between treatment group.** Relative abundances of *C. difficile* transcript for specific genes of interest. **(A)** Transcription for select genes from the *C. difficile* sporulation pathway with the greatest variation in expression between the conditions tested. **(B)** Relative abundances of transcript for genes that encode effector proteins from the *C. difficile* pathogenicity locus. **(C)** Transcript abundances for genes associated with quorum sensing in *C. difficile*. **(D)** Transcript relative abundance of select sigma factors which expression or activity is influenced by environmental metabolite concentrations. Asterisks (\*) indicate genes from which transcript was undetectable.

**Supplementary Figure 4 | Additional explanation for Figure 2 interpretation.** Relative abundance of transcription for *C. difficile* 630 genes during infection across the 3 antibiotic pretreatment models used during this study. Points that are located closer to a corner are more highly transcribed in the condition associated with that corner compared to the others. As this shows a 3-dimensional data set in 2 dimensions, there is an amount of distortion proximal to each corner. Simply put for points that are nearer to an edge, a greater percentage of their total transcription was contributed by *C. difficile* colonizing those mice. **(A)** This point represents the transcription for a gene that is overrepresented in cefoperazone-treated mice. **(B)** This point represents a gene in which transcripts are equally detectable in all 3 conditions. **(C)** Transcripts for this gene are only underrepresented in only cefoperazone-treated mice, and are equally detectable in clindamycin and streptomycin-treated animals.

**Supplementary Figure 5 | Within-group sample variance for community-level data with replication.** **(A)** Distribution of within-group OTU abundance sample variance from 16S rRNA gene sequencing. **(B)** Distribution of within-group scaled intensities from untargeted metabolomic analysis. The between-group variances for OTU and metabolomic data were all greater than one, indicating that there were low levels of variation between samples.

**Supplementary Figure 6 | Relative *in vivo* concentrations of metabolites highlighted through network-based analysis in uninfected mice.** Asterisks indicate those treatment groups where a metabolite had a significantly different metabolite concentration compared to the no antibiotics SPF mice.

**Supplementary Figure 7 | Change in *in vivo* concentrations following *C. difficile* infection of additional select metabolites.** Asterisks indicate those treatment groups where a metabolite had a significantly different metabolite concentration in *C. difficile* colonized and non-colonized conditions.



**Supplementary Table 1 | Sets of genes included in Figure 2 with normalized abundances and annotations.**

**Supplementary Table 2 | Topology metrics for enzyme and metabolite nodes in the *C. difficile* str. 630 metabolic network.**

**Supplementary Table 3 | All metabolites with significant importance scores in each colonized condition.**

**Supplementary Table 4 | Growth curve analysis for each tested carbon source.**

~~**Supplementary Table 5 | Outliers from generalized linear models of combined metabolome and metabolic network output analysis.**~~


### References
