---
title: Clostridium difficile colonizes alternative nutrient niches during infection across distinct murine gut environments
csl: bmc.csl
output: 
 word_document:
    keep_md: true
    reference_docx: manuscript_format.docx
bibliography: references.bibtex
---



**Authors:** Nielson T. Baxter^2^, Charles C. Koumpouras^1^, Mary A.M. Rogers^2^, Mack T. Ruffin IV^3^, and Patrick D. Schloss^1\*^


**Affiliations:**  
^1^Department of Microbiology and Immunology, University of Michigan, Ann Arbor, Michigan.  
^2^Department of Internal Medicine, University of Michigan, Ann Arbor, Michigan.  
^3^3Department of Family and Community Medicine, Penn State Hershey medical Center, Herhsey, Pennsylvania    
^\*^Corresponding author: pschloss@umich.edu

**Email addresses:**  
NTB: ntbaxter@umich.edu  
CCK: chkoumpo@umich.edu  
MAMR: maryroge@med.umich.edu  
MTR: mruffin@hmc.psu.edu  
PDS: pschloss@umich.edu  


```{r startup, echo=F, message=F, warning=F}
deps <- c("rmarkdown","AUCRF","pROC");
for (dep in deps){
  if (dep %in% installed.packages()[,"Package"] == FALSE){
    install.packages(as.character(dep), quiet=TRUE);
  }
  library(dep, verbose=FALSE, character.only=TRUE)
}

pval <- function(p){
  if(p < 0.001){p <- 'p<0.001'}
  else{p <- sprintf('p=%.2f', p)}
  return(p)
}

setwd('~/Desktop/Baxter_FITs_2016/')
meta <- read.delim('data/metadata.tsv', header=T, sep='\t')
shared <- read.table('data/fit.final.an.unique_list.0.03.subsample.0.03.filter.shared', header=T)
shared <- shared[,-c(1,3)]

meta$lesion <- c()
meta$lesion[meta$dx=='normal'] <- 0
meta$lesion[meta$dx!='normal'] <- 1
meta$lesion <- as.factor(meta$lesion)

```


###Abstract  
**Background:** There is a significant demand for colorectal cancer (CRC) screening methods that are noninvasive, inexpensive, and capable of accurately detecting early stage tumors. It has been shown that models based on the gut microbiota can complement the fecal occult blood test and fecal immunochemical test (FIT). However, a barrier to microbiota-based screening is the need to collect and store a patient's stool sample.  
**Results:** Using stool samples collected from 404 patients we tested whether the residual buffer containing resuspended feces in FIT cartridges could be used in place of intact stool samples. We found that the bacterial DNA isolated from FIT cartridges largely recapitulated the community structure and membership of patients' stool microbiota and that the abundance of bacteria associated with CRC were conserved. We also found that models for detecting CRC that were generated using bacterial abundances from FIT cartridges were equally predictive as models generated using bacterial abundances from stool.  
**Conclusions:** These findings demonstrate the potential for using residual buffer from FIT cartridges in place of stool for microbiota-based screening for CRC. This may reduce the need to collect and process separate stool samples and may facilitate combining FIT and microbiota-based biomarkers into a single test. Additionally, FIT cartridges could constitute a novel data source for studying the role of the microbiome in cancer and other diseases.

**Keywords:** colorectal cancer, gut microbiome, microbiota, fecal immunochemical test, random forest

###Background  
Although colorectal cancer (CRC) mortality has declined in recent decades, it remains the second leading cause of death among cancers in the United States [@siegel2014colorectal]. Early detection of CRC is critical since patients whose tumors are detected at an early stage have a greater than 90% chance of survival [@siegel2014colorectal]. However more than a third of Americans for whom screening is recommended do not adhere to screening guidelines [@centers2013vital]. The high cost and invasive nature of procedures, such as colonoscopy and sigmoidoscopy are barriers for many people [@hsia2000importance;@jones2010barriers]. Some do not consider these invasive test as a primary screening option, but they are in the United States [@bibbins2016screening]. Unfortunately non-invasive tests, such as the guaiac fecal occult blood test (gFOBT), fecal immunochemical test (FIT), and the multitarget DNA test fail to reliably detect adenomas [@hundt2009comparative;@imperiale2014multitarget] (e.g., sensitivity for nonadvanced adenomas is 7.6% for FIT and 17.2% for the DNA test). Thus, there is a need for novel non-invasive screening methods with improved sensitivity for early stage colonic lesions.

Several studies have demonstrated the potential for the gut microbiota to be used to detect CRC [@zackular2014human;@zeller2014potential;@yu2015metagenomic;@baxter2016microbiota]. Moreover, we and others have shown that combining microbiota-analysis with conventional diagnostics, like gFOBT and FIT, can significantly improve the detection of colonic lesions over either method by itself [@zackular2014human;@zeller2014potential;@baxter2016microbiota]. One limitation of microbiota-based CRC screening is the need to collect and process separate stool samples for microbiota characterization. Given the widespread use of FIT to collect specimens for screening, the ability to use the same sample for microbiota characterization could make processing more efficient and less expensive. We hypothesized that the small amount of fecal material contained in FIT sampling cartridges was sufficient to perform both hemoglobin quantification and microbiota characterization. To test this hypothesis, we isolated bacterial DNA from the residual buffer of OC-Auto® FIT cartridges (Polymedco Inc.) that had already been used for quantifying fecal hemoglobin concentrations. We then compared the bacterial composition of the FIT cartridge to that of DNA isolated directly from a patient's stool sample and assessed the ability of FIT cartridge-derived DNA to be used for microbiota-based CRC screening.


###Methods
**Study Design / Diagnoses / Stool Collection.** Stool samples were obtained through the Great Lakes-New England Early Detection Research Network. The study was a multi-center, prospective, cross-sectional cohort validation study planned for 600 participants. The planned sample size was 200 participants with colorectal cancer, 200 participants with colorectal adenomas, and 200 participants with normal colonoscopies. The baseline visit was done either prior to a scheduled colonoscopy or after a colonoscopy was done. If a subject was suspected of having a colon adenocarcinoma or an adenoma, the baseline samples was collected before the colonoscopy or exposure to colonic preps so the lesions are present when stool collection is done. Normal controls were recruited at the time of their scheduled colonoscopy, prior to their colonoscopy, or within 4 months after completing a screening colonoscopy indicating no adenomas or colorectal cancer to provide a set of samples. The normal patients stool samples were collected before colonic preps or at least four weeks after the colonoscopy was completed. Patients were asymptomatic, at least 18 years old, willing to sign informed consent, able to tolerate removal of 58 mL of blood, and willing to collect a stool sample. Patient age at the time of enrollment ranged from 29 to 89 with a median of 60 years. Patients were excluded if they had undergone surgery, radiation, or chemotherapy for current CRC prior to baseline samples or had inflammatory bowel disease, known hereditary non-polyposis CRC, or familial adenomatous polyposis. Patient diagnoses were determined by colonoscopic examination and histopathological review of any biopsies taken. Colonoscopies were performed and fecal samples were collected in four locations: Toronto (Ontario, Canada), Boston (Massachusetts, USA), Houston (Texas, USA), and Ann Arbor (Michigan, USA). All participants collected a whole evacuated stool in a hat with no preservatives. Stool samples were collected in packed in ice, shipped to a processing center via next day delivery and stored at -80˚C. Fecal material for FIT was collected from frozen stool aliquots using OC-Auto® FIT sampling bottles (Polymedco Inc.). FIT cartridges were store at 4˚C for up to 48 hours prior to hemoglobin quantification using an OC-Auto Micro 80 automated system (Polymedco Inc.). FIT cartridges were then stored at -20˚C until being thawed for DNA extraction. The University of Michigan Institutional Review Board approved this study, and all subjects provided informed consent.


**16S rRNA gene sequencing.** Processed FIT samples were thawed, and 100 µl of buffer were withdrawn by pipette for DNA extraction. DNA was isolated from the FIT cartridge buffers and the matching stool samples from which they were collected using the PowerSoil-htp 96 Well Soil DNA isolation kit (MO BIO Laboratories) and an epMotion 5075 automated pipetting system (Eppendorf). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers and sequenced as described previously using an Illumina MiSeq sequencer [@miseq]. The 16S rRNA gene sequences were curated using the mothur software package, as described previously [@mothur;@miseq]. Curated sequences were clustered into operational taxonomic units (OTUs) using a 97% similarity cutoff with the average neighbor clustering algorithm. Sequences were classified using a naive Bayesian classifier trained against a 16S rRNA gene training set provided by the Ribosomal Database Project (RDP) [@wang2007naive]. Species-level classifications for OTUs of interest were determined by using blastn to compare the predominant sequence within each OTU to the NCBI 16S rRNA database. The putative species was only reported for OTUs with greater than 99% sequence identity to a single species in the database; otherwise the consensus RDP classification was used.


**Statistical Methods.** All statistical analyses were performed using R (v.3.2.0). Random forest models were generated using the AUC-RF algorithm for feature reduction and maximizing model performance [@calle2011auc]. The most predictive OTUs were determined based on mean decrease in accuracy when removed from the model. The area under the curve (AUC) of receiver operator characteristic (ROC) curves were compared using the method described by DeLong et al. [@delong1988comparing] as implemented in the pROC R package [@robin2011proc]. There was sufficient power (80%) to detect a 0.081 difference in AUC with alpha set at 0.05.


###Results

```{r distance_density, echo=F, warning=F, message=F, results='hide', cache=T}
source('code/read.dist.R')
dist <- read.dist('data/fit.final.an.unique_list.thetayc.0.03.lt.ave.dist')

fits <- grep('fit', row.names(dist), value=T)
stools <- grep('fit', row.names(dist), value=T, invert=T)

pairs <- rep(NA, length(fits))
for(i in 1:length(fits)){
  pairs[i]<-1-as.numeric(dist[fits[i],stools[i]])
}  

stool2stool <- dist[stools,stools]
stool2stool <- 1-as.numeric(stool2stool[upper.tri(stool2stool, diag=F)])

fit2fit <- dist[fits,fits]
fit2fit <- 1-as.numeric(fit2fit[upper.tri(fit2fit, diag=F)])

stool2fit <- 1-as.numeric(as.vector(unlist(dist[stools,fits])))


tiff('results/figure1.tiff', width=9, height=4.5, units='in', res=150)
layout(matrix(c(2,1), nrow=1), widths=c(1,1))
par(mar=c(3.5,2,2,1), mgp=c(2,0.5,0), oma=c(0,0,0,0))
plot(density(pairs), type='l',lwd=3, col='red', ylim=c(0,4), yaxt='n', xlab='Community Similarity (1-thetaYC)', main='', las=1, tck=-0.02)
points(density(stool2stool), type='l',lwd=3, col='orange')
points(density(fit2fit), type='l',lwd=3, col='blue')
points(density(stool2fit), type='l',lwd=3, col='green4')
legend('topright', c('Stool/FIT pairs','Between FITs & Stool','Between Stools','Between FITs'), col=c('red','green4','orange','blue'), lwd=3, bty='n', cex=0.8)
mtext('Density',side=2, at=2, line=0.5)
points(0.7,max(density(pairs)$y)*1.07, pch=8)
mtext('B', at=-0.25, side=3, line=0.1, font=2, cex=2)

thyc_p <- ks.test(pairs, fit2fit, alternative='less')$p.value


dist <- read.dist('data/fit.final.an.unique_list.sharedsobs.0.03.lt.ave.dist', diag=0)

pairs <- rep(NA, length(fits))
for(i in 1:length(fits)){
  pairs[i]<-as.numeric(dist[fits[i],stools[i]])
}  

stool2stool <- dist[stools,stools]
stool2stool <- as.numeric(stool2stool[upper.tri(stool2stool, diag=F)])

fit2fit <- dist[fits,fits]
fit2fit <- as.numeric(fit2fit[upper.tri(fit2fit, diag=F)])

stool2fit <- as.numeric(as.vector(unlist(dist[stools,fits])))


par(mar=c(3.5,2.5,2,0.5))
plot(density(pairs), type='l', lwd=3, col='red', ylim=c(0,0.025), yaxt='n', xlab='Number of shared OTUs', ylab='', main='', las=1)
points(density(stool2stool), type='l',lwd=3, col='orange')
points(density(fit2fit), type='l',lwd=3, col='blue')
points(density(stool2fit), type='l',lwd=3, col='green4')
legend('topright', c('Stool/FIT pairs','Between FITs & Stool','Between Stools','Between FITs'), col=c('red','green4','orange','blue'), lwd=3, bty='n', cex=0.8)
points(median(pairs),max(density(pairs)$y)*1.1, pch=8)
mtext('Density',side=2, at=0.0125, line=0.5)
mtext('A', at=-30, side=3, line=0.1, font=2, cex=2)
dev.off()

sobs_p <- ks.test(pairs, fit2fit, alternative='less')$p.value

```

```{r mantel, echo=F}
mantel <- read.table('data/fit_only.an.thetayc.0.03.lt.ave.mantel', header=T)
mantel_p <- mantel$pValue
mantel_r <- mantel$Mantel
```

```{r trimTax, engine='sh', echo=F, results='hide'}
python code/trimTax.py data/fit.final.tx.1.cons.taxonomy data/fit.phyla.tax 5
python code/trimTax.py data/fit.final.an.unique_list.0.03.cons.taxonomy data/fit.otus.tax 1
```


```{r compareGenera, echo=F, results='hide', warning=F, cache=T}
genera <- read.table('data/fit.final.tx.1.subsample.shared', header=T)
num_seqs <- sum(genera[3,4:ncol(genera)])

stool_genera <- genera[-grep('fit',genera$Group), grep('Otu[0-9]',colnames(genera))]
stool_gen_avg <- apply(stool_genera, 2, mean)/num_seqs + 1e-6

fit_genera <- genera[grep('fit',genera$Group), grep('Otu[0-9]',colnames(genera))]
fit_gen_avg <- apply(fit_genera, 2, mean)/num_seqs + 1e-6

fit_gen_avg <- fit_gen_avg[order(stool_gen_avg, decreasing=T)]
stool_gen_avg <- sort(stool_gen_avg, decreasing=T)

taxonomy <- read.delim('data/fit.phyla.tax', header=T, row.names=1, sep='\t')
phyla <- taxonomy[names(fit_gen_avg),'Classification']
levels(phyla)[!(levels(phyla)%in%c('Bacteroidetes','Firmicutes','Proteobacteria','Actinobacteria','Verrucomicrobia','Fusobacteria'))] <- 'Other'

tiff('results/figure2.tiff', height=5, width=10, units='in', res=150)
layout(matrix(c(1,1,1,1,2,4,3,5),nrow=2))
par(mar=c(2,3,1.5,2), mgp=c(2,0.5,0), oma=c(2,1,1,1))
plot(stool_gen_avg, fit_gen_avg, log='xy', xlim=c(1e-6,1), ylim=c(1e-6,1), yaxt='n', xaxt='n', pch=21, bg=c('grey','purple','red','blue','green','yellow','orange2')[phyla], ylab='', xlab='', cex=1.3)
abline(0,1, lty=3)
axis(1, at=c(1e-6,1e-5,1e-4,1e-3,1e-2,1e-1,1), labels=c(0,0.001,0.01,0.1,1,10,100))
axis(2, at=c(1e-6,1e-5,1e-4,1e-3,1e-2,1e-1,1), labels=c(0,0.001,0.01,0.1,1,10,100))
legend('bottomright', c('Bacteroidetes','Firmicutes','Proteobacteria','Actinobacteria','Verrucomicrobia','Fusobacteria','Others'), pch=21, pt.bg=c('red','blue','yellow','purple','orange2','green','grey'), pt.cex=1.3)

mtext('A', at=3e-7, side=3, line=0, font=2, cex=2)
mtext('Average relative abundance in Stool (%)', side=1, line=2, cex=0.8)
mtext('Average relative abundance in FIT cartridge (%)', side=2, line=2, cex=0.8)
mtext('B', side=3, line=0, at=5, font=2, cex=2)



gen_corr <- cor.test(stool_gen_avg,fit_gen_avg, method='spearman')
gen_100_corr <- cor.test(stool_gen_avg[1:100],fit_gen_avg[1:100], method='spearman')

# Figure 2B: OTU Scatterplots
stool_shared<-shared[meta$type=='stool',]
fit_shared<-shared[meta$type=='fit',]


otus <- c('Otu000202','Otu000070','Otu000548','Otu000393')
taxa <- c(expression(italic('Fusobacterium nucleatum')),expression(italic('Porphyromonas asaccharolytica')),expression(italic('Peptostreptococcus stomatis')),expression(italic('Parvimonas micra')))


stool_abunds <- stool_shared[,otus]/3879 + 0.0001
fit_abunds <- fit_shared[,otus]/3879 + 0.0001

otu_pvals <- c() 
otu_corrs <- c()
par(mar=c(2,2,1.5,0.5), mgp=c(2,0.5,0))
for(i in 1:length(otus)){
  plot(log10(jitter(stool_abunds[,otus[i]], amount=0.00002)), log10(jitter(fit_abunds[,otus[i]], amount=0.00002)), xaxt='n', yaxt='n', ylab='', xlab='', pch=21, cex=1, bg=c('orange','red','royalblue1')[meta$dx], main=taxa[i], cex.main=1, xlim=c(-4,-0.5), ylim=c(-4,-0.5))
  abline(0,1, lty=3)
  axis(1, at=c(-4,-3,-2,-1), labels=c(0,0.1,1,10) )
  axis(2, at=c(-4,-3,-2,-1), labels=c(0,0.1,1,10) )
  otu_pvals[i] <- cor.test(stool_shared[,otus[i]], fit_shared[,otus[i]], method='spearman')$p.value
  otu_corrs[i] <- cor.test(stool_shared[,otus[i]], fit_shared[,otus[i]], method='spearman')$estimate
}
legend('bottomright', legend=c('Normal','Adenoma','Cancer'), pch=21, pt.cex=1, pt.bg=c('royalblue1', 'orange','red'))


mtext('Relative abundance in Stool (%)', side=1, at=0.75 ,line=0, outer=T, cex=0.8)
mtext('Relative abundance in FIT cartridge (%)', side=2, line=-37, outer=T, cex=0.8)


dev.off()

```

DNA was isolated and 16S rRNA gene sequencing was performed on stool aliquots and the residual buffer of paired OC-Auto® FIT sampling cartridges from 404 patients. Among these patients, 101 had CRC, 162 had adenomas, and 141 had no colonic lesions. First, we tested whether the bacterial community profiles from FIT cartridges recapitulated their stool counterparts. First, we compared the number of OTUs shared within FIT/stool pairs from the same patient to the number of OTUs shared between patients (Fig. 1A). FIT cartridges and stool from the same patient (red line) had significantly more bacterial populations in common than those taken from different patients (`r pval(sobs_p)`, two-sample Kolmogorov-Smirnov test), indicating that community membership was conserved within patients across stool and FIT cartridges. Second, we calculated the similarity in community structure between samples using 1-thetaYC index [@yue2005similarity]. This metric compares the presence or absence of bacterial populations and their relative abundance. The bacterial community structure of stool and FIT samples from the same patient (red line) were significantly more similar to each other than to stool or FIT from other patients (Fig. 1B, `r pval(thyc_p)`). Finally, we used a Mantel test to determine whether the patient-to-patient thetaYC distances among stool samples were correlated with the patient-to-patient thetaYC distances among FIT cartridges. We found that there was a significant correlation (Mantel test r=`r signif(mantel_r, 3)`, `r pval(mantel_p)`), suggesting that the inter-patient variation in community structure between the stool samples of patients was conserved in samples from FIT cartridges . 

Next, we observed a significant correlation between the abundance of each genus in the paired FIT cartridge and stool samples (Fig. 2A, `r sprintf('Spearman rho: %.3f', gen_corr$estimate)`, `r pval(gen_corr$p.value)`). This suggested that the abundance of bacterial genera was conserved. This correlation was especially strong when comparing the 100 most abundant genera from stool (`r sprintf('Spearman rho: %.3f', gen_100_corr$estimate)`, `r pval(gen_100_corr$p.value)`). Several bacterial species have been repeatedly associated with CRC, including _Fusobacterium nucleatum_, _Porphyromonas asaccharolytica_, _Peptostreptococcus stomatis_, and _Parvimonas micra_ [@warren2013co;@zeller2014potential;@yu2015metagenomic;@baxter2016microbiota].  As expected, the abundance of these species in stool was significantly correlated with their abundance in matched FIT cartridges (all `r pval(max(otu_pvals))`, Spearman rho ≥`r sprintf('%.3f', min(otu_corrs))`)(Fig. 2B). We observed some biases in the abundance of certain taxa. In particular, the genus *Pantoea* was detected in `r sum(fit_genera$Otu127>0)` of the 404 FIT cartridges with an average abundance of `r sprintf('%.1f%%', fit_gen_avg['Otu127']*100)`, but was only detected in `r sum(stool_genera$Otu127>0)` stool sample. The genus *Helicobacter* was detected in `r sum(fit_genera$Otu032>0)` FIT cartridges, but only `r sum(stool_genera$Otu032>0)` stool samples. Likewise several genera of *Actinobacteria* were more abundant in stool samples compared to FIT. Notwithstanding these few exceptions, the abundance of the vast majority of genera were well conserved between stool and FIT cartridges. Overall, these findings suggested that that the overall bacterial community structure and the abundance of specific taxa in FIT cartridges and stool were similar.


```{r compare_models, echo=F, cache=T, warning=F, message=F, results='hide', fig.height=4, fig.width=8}

stool_data<-merge(meta, stool_shared, by.x='sample',by.y='Group')
fit_data<-merge(meta, fit_shared, by.x='sample',by.y='Group')
stool_data <- stool_data[,c('lesion',colnames(stool_data)[grep('Otu[0123456789]', colnames(stool_data))])]
fit_data <- fit_data[,c('lesion',colnames(stool_data)[grep('Otu[0123456789]', colnames(stool_data))])]
dx <- meta$dx[1:404]

### Lesion vs Normal models
set.seed(021316)
fit_les_model <- AUCRF(lesion~., data=fit_data, pdel=0.05, ntree=500, ranking='MDA')
fit_les_probs <- predict(fit_les_model$RFopt, type='prob')[,2]
fit_les_probs <- (fit_les_probs-0.9*min(fit_les_probs))/(1-min(fit_les_probs)) #normalization to spread out probabilities
fit_les_roc <- roc(fit_data$lesion~fit_les_probs)

set.seed(021316)
stool_les_model <- AUCRF(lesion~., data=stool_data, pdel=0.05, ntree=500, ranking='MDA')
stool_les_probs <- predict(stool_les_model$RFopt, type='prob')[,2]
stool_les_probs <- (stool_les_probs-0.9*min(stool_les_probs))/(1-min(stool_les_probs)) #normalization to spread out probabilities
stool_les_roc <- roc(stool_data$lesion~stool_les_probs)

# Figure layout
tiff('results/figure3.tiff', width=9, height=8, units='in', res=150)
layout(matrix(c(3,1,4,2),nrow=2))
par(mar=c(3,3.5,2,1), mgp=c(2,0.5,0), oma=c(0,1.5,0,0))

### Compare ROC curves
plot(c(1,0),c(0,1), type='l', lty=2, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity', las=1)
plot(fit_les_roc, add=T, lwd=2, col='purple', lty=1)
plot(stool_les_roc, add=T, lwd=2, col='green3', lty=1)
legend('bottomright', c(sprintf('FIT cartridge: AUC=%.3f',fit_les_roc$auc),sprintf('Stool: AUC=%.3f',stool_les_roc$auc)), lwd=2, col=c('purple','green3'), bty='n', lty=c(1,1))
mtext('C', at=1.1, side=3, line=0.1, font=2, cex=2)


### Compare model results - scatter plot
plot(stool_les_probs,fit_les_probs, bg=c('orange','red','royalblue1')[dx], pch=21, ylab='Probability of Lesion (FIT cartridge)', xlab='Probability of Lesion (Stool)', ylim=c(0,1), xlim=c(0,1), las=1)
legend('topleft', c('Cancer','Adenoma','Normal'), pch=21, pt.bg=c('red','orange','royalblue1'))
mtext('D', at=-0.1, side=3, line=0.1, font=2, cex=2)

### Cancer vs normal models
stool_data <- stool_data[dx!='adenoma',]
fit_data <- fit_data[dx!='adenoma',]

set.seed(021316)
fit_canc_model <- AUCRF(lesion~., data=fit_data, pdel=0.05, ntree=500, ranking='MDA')
fit_canc_probs <- predict(fit_canc_model$RFopt, type='prob')[,2]
fit_canc_roc <- roc(fit_data$lesion~fit_canc_probs)

set.seed(021216)
stool_canc_model <- AUCRF(lesion~., data=stool_data, pdel=0.05, ntree=500, ranking='MDA')
stool_canc_probs <- predict(stool_canc_model$RFopt, type='prob')[,2]
stool_canc_roc <- roc(stool_data$lesion~stool_canc_probs)

### Compare ROC curves
plot(c(1,0),c(0,1), type='l', lty=2, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity', las=1)
plot(fit_canc_roc, add=T, lwd=2, col='purple', lty=1)
plot(stool_canc_roc, add=T, lwd=2, col='green3', lty=1)
legend('bottomright', c(sprintf('FIT cartridge: AUC=%.3f',fit_canc_roc$auc),sprintf('Stool: AUC=%.3f',stool_canc_roc$auc)), lwd=2, col=c('purple','green3'), bty='n', lty=1)
mtext('A', at=1.1, side=3, line=0.1, font=2, cex=2)

### Compare model results - scatter plot
plot(stool_canc_probs,fit_canc_probs, bg=c('royalblue1','red')[stool_data$lesion], pch=21, ylab='Probability of Lesion (FIT cartridge)', xlab='Probability of Lesion (Stool)', ylim=c(0,1), xlim=c(0,1), las=1)
legend('topleft', c('Cancer','Normal'), pch=21, pt.bg=c('red','royalblue1'))
mtext('B', at=-0.1, side=3, line=0.1, font=2, cex=2)

mtext('Cancer vs. Normal', las=0, side=2, at=0.77, outer=T, adj=0.5, line=0, font=2, cex=1)
mtext('Any lesion vs. Normal', las=0, side=2, at=0.27, outer=T, adj=0.5, line=0, font=2, cex=1)

dev.off()

les_corr <- cor.test(fit_les_probs,stool_les_probs, method='spearman')
canc_corr <- cor.test(fit_canc_probs,stool_canc_probs, method='spearman')
```


We tested whether the bacterial relative abundances we observed from FIT cartridges could be used to differentiate healthy patients from those with carcinomas using random forest models as we did previously using intact stool samples [@baxter2016microbiota]. Using DNA from the FIT cartridge, the optimal model utilized `r fit_canc_model$Kopt` OTUs and had an AUC of `r sprintf('%.3f', fit_canc_roc$auc)` (Fig. 3A). There was not a significant difference in the AUC for this model and the model based on DNA isolated directly from stool, which used `r stool_canc_model$Kopt` OTUs and had an AUC of `r sprintf('%.3f',stool_canc_roc$auc)` (`r sprintf('p=%.2f', roc.test(fit_canc_roc, stool_canc_roc)$p.value)`). Furthermore, the probabilities of individuals having lesions were correlated between the models generated using DNA isolated from the FIT cartridges and stool samples (`r sprintf('Spearman rho: %.3f', canc_corr$estimate)`, `r pval(canc_corr$p.value)`, Fig. 3B). We also generated random forest models for differentiating healthy patients from those with any type of lesions (i.e. adenoma or carcinoma). There was not a significant difference in AUC between the stool-based model with `r stool_les_model$Kopt` OTUs (AUC=`r sprintf('%.3f', stool_les_roc$auc)`) and the FIT cartridge-based model with `r fit_les_model$Kopt` OTUs (AUC=`r sprintf('%.3f', fit_les_roc$auc)`, `r sprintf('p=%.2f', roc.test(fit_les_roc,stool_les_roc)$p.value)`, Fig. 3C). Again, the probabilities of individuals having lesions according to the two models were significantly correlated (`r sprintf('Spearman rho: %.3f', les_corr$estimate)`, `r pval(les_corr$p.value)` Fig. 3D). These findings demonstrated that models based on bacterial DNA from FIT cartridges were as as predictive as models based on DNA isolated directly from stool.



###Discussion
Bacterial DNA isolated from the residula buffer of FIT cartridges recapitulated the community structure and membership of patients' stool microbiota. FIT/stool pairs collected from the same patient were significantly more similar to each other than samples from different patients and the inter-patient differences in stool microbiota structure were conserved in FIT cartridge-derived microbiota.  More importantly, random forest models generated using bacterial abundances from FIT cartridge-derived and stool-derived DNA were equally predictive for differentiating healthy patients from those with adenomas and carcinomas.

Sinha et al. compared a variety of sampling and storage methods for fecal samples to be used for microbiome analyses [@sinha2015collecting]. They found reproducible biases according to sampling method and time at ambient temperature. Likewise, we observed biases in the abundance certain bacterial populations in FIT cartridges compared stool. For example, an OTU associated with _Pantoea_ was found in `r sprintf('%.1f', 100*sum(fit_genera$Otu127>0)/nrow(fit_genera))`% of FIT cartridge samples and only `r sprintf('%.1f', 100*sum(stool_genera$Otu127>0)/nrow(stool_genera))`% of stool samples. It is likely that because the biomass contained in the FIT cartridges is considerably lower than that in stool, the analysis was more sensitive to contaminants in our reagents or the FIT cartridges themselves [@salter2014reagent]. This seems to be the case for genus _Pantoea_, since a sequence classified as _Pantoea_ was detected in several uninoculated FIT cartridges used as negative controls. It is also possible that storage conditions could have played a role in biasing the relative abundances of certain genera. The feces in the FIT cartridges spent more time exposed to ambient temperatures while being analyzed for hemoglobin concentration. Therefore it is possible that certain bacterial populations, especially aerobes, were able to grow Alternatively, there could have been a difference in DNA extraction efficiency between the stool samples and FIT cartridge buffer. Actinobacteria, which were overrepresented in stool samples, require mechanical lysis for DNA extraction [@walker201516s]. It is possible that the particulates in stool allowed for more efficient mechanical lysis of Actinobacteria cells compared to the FIT cartridge buffer. Regardless, these biases had no effect on the ability to detect CRC from FIT cartridge-derived DNA, as the random forest feature selection process did not select these populations.

Colonoscopy, rather than stool screening, remains the more effective CRC screening approach. However, non-invasive stool tests, including those based on microbiota, might have future utility in directing high risk subjects towards a colonoscopy, increasing the overall performance of CRC screening and reducing morbidity and mortality. The ability to assay hemoglobin qunatification and microbiota-based bioimarkers with single sampling kit would reduce cost and improve efficiency. It would also facilite combining results from the two methods into single model for predicting the presence of lesions, which has been shown to improve sensitivity over either method alone [@baxter2016microbiota]. Incoporating other host-associated stool biomarkers and clinical data could further improve the accuracy of such a model.

###Conclusions  
The findings from this study demonstrate the potential for using DNA from FIT cartridges for microbiota-based screening. This could reduce the need to collect and process separate stool samples, decreasing the cost of screening. It is also possible to use FIT cartridges rather than separate stool samples for future studies on the role of the gut microbiota and cancer. Samples collected from patients who undergo annual FIT screening could be used to monitor temporal changes in a patient's microbiota, making it possible to detect shifts toward a disease-associated microbiota. Since FIT cartridges are currently used for CRC screening, our findings may facilitate large-scale validations of microbiota-based screening methods.

 
**Abbreviations:**  
FIT: fecal immunochemical test  
gFOBT: guaic fecal occult blood test  
OTU: operational taxonomic unit  
AUC: area under the curve  
ROC curve: reciever operating characteristic curve  


###Declarations 

**Ethics approval and consent to participate:** The University of Michigan Institutional Review Board approved this study, and all subjects provided informed consent. This study conformed to the guidelines of the Helsinki Declaration.

**Consent for publication:** Not applicable. 

**Availability of data and materials:** Raw sequencing data patient metadata are available through the NCBI Sequence Read Archive (SRP062005 for stool samples and metadata, SRP073172 for FIT cartridges). The data processing steps for going from the raw sequence data to the final manuscript is available at http://www.github.com/SchlossLab/Baxter_FITs_Microbiome_2016.
**Consent for publication:** Not applicable.


**Competing interests:** The authors declare that they have no competing interests.  

**Funding:** This study was supported by funding from the National Institutes of Health to P. Schloss (R01GM099514, P30DK034933) and to the Early Detection Research Network (U01CA86400).

**Author Contributions:** PDS, MTR, MAMR, and NTB were involved in the conception and design of the study. NTB and CCK performed DNA extractions and 16S rRNA gene sequencing. NTB analyzed the data. All authors interpreted the data. NTB and PDS wrote the manuscript. All authors reviewed and revised the manuscript.

**Acknowledgements:** The authors thank the Great Lakes-New England Early Detection Research Network for providing the fecal samples that were used in this study. 


###Figures  
 
**Figure 1. Bacterial community structure from FIT cartridge recapitulates stool.** Density plots showing distribution of the number of shared OTUs (A) and community similarity (B) between groups of samples (* p<0.001 two-sample Kolmogorov-Smirnov Test).

**Figure 2. Bacterial populations conserved between stool and FIT cartridge.**  (A) Scatterplot of the average relative abundance of each bacterial genus in stool and FIT cartridges colored by phylum. (B) Scatterplots of the relative abundances of 4 species frequently associated with CRC. All correlations were greater than `r floor(min(otu_corrs)*100)/100` (all `r pval(max(otu_pvals))`).
 
**Figure 3. Microbiota-based models from FIT cartridge DNA are as predictive as models from stool.** (A) ROC curves for distinguishing healthy patients from those with cancer using microbiota-based random forest models using DNA from FIT cartridges or stool. (B) Probability of having cancer for each patient according to microbiota-based models from A. (C) ROC curves for distinguishing patients with adenomas or carcinomas from healthy patients using microbiota-based random forest models using DNA from FIT cartridges or stool. (D) Probability of having a lesion for each patient based on the models from C.


###References
